{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "presidential-figure",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "tutorial-extra",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import sys\n",
    "\n",
    "module_path = os.path.abspath(os.path.join('..'))\n",
    "if module_path not in sys.path:\n",
    "    sys.path.append(module_path)\n",
    "\n",
    "# FIX ME: Root dir as argument\n",
    "root_dir = \"../../data/picklerCheckpoints/\"\n",
    "is_compressed = True;\n",
    "\n",
    "os.environ[\"CUDA_VISIBLE_DEVICES\"] = \"-1\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "portuguese-creature",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tensor Flow Version: 2.1.0\n",
      "Keras Version: 2.2.4-tf\n",
      "\n",
      "Python 3.7.9 (default, Aug 31 2020, 17:10:11) [MSC v.1916 64 bit (AMD64)]\n",
      "Pandas 1.2.1\n",
      "Scikit-Learn 0.24.1\n",
      "GPU is NOT AVAILABLE\n"
     ]
    }
   ],
   "source": [
    "# What version of Python do you have?\n",
    "import sys\n",
    "\n",
    "import tensorflow.keras\n",
    "import pandas as pd\n",
    "import sklearn as sk\n",
    "import tensorflow as tf\n",
    "\n",
    "print(f\"Tensor Flow Version: {tf.__version__}\")\n",
    "print(f\"Keras Version: {tensorflow.keras.__version__}\")\n",
    "print()\n",
    "print(f\"Python {sys.version}\")\n",
    "print(f\"Pandas {pd.__version__}\")\n",
    "print(f\"Scikit-Learn {sk.__version__}\")\n",
    "gpu = len(tf.config.list_physical_devices('GPU'))>0\n",
    "print(\"GPU is\", \"available\" if gpu else \"NOT AVAILABLE\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "younger-dallas",
   "metadata": {},
   "outputs": [],
   "source": [
    "from tensorflow.python.client import device_lib\n",
    "\n",
    "def get_available_devices():\n",
    "    local_device_protos = device_lib.list_local_devices()\n",
    "    return [x.name for x in local_device_protos if x.device_type == 'GPU' or x.device_type == 'CPU']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "educational-commission",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['/device:CPU:0']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_available_devices()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "according-bahamas",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import tensorflow.python.keras.backend as K\n",
    "# import tensorflow.compat.v1 as tf\n",
    "# tf.disable_v2_behavior() \n",
    "# jobs = 6 # number of cores\n",
    "# config = tf.ConfigProto(intra_op_parallelism_threads=jobs,\n",
    "#                      inter_op_parallelism_threads=jobs,\n",
    "#                      allow_soft_placement=True,\n",
    "#                      device_count={'CPU': jobs})\n",
    "# session = tf.Session(config=config)\n",
    "# K.set_session(session)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "julian-drove",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Importing Jupyter notebook from D:\\Academics\\MLRepoNew\\ML2020-Uber-Data-Analysis\\commons\\utility\\pickle_util.ipynb\n",
      "Importing Jupyter notebook from D:\\Academics\\MLRepoNew\\ML2020-Uber-Data-Analysis\\Preprocessing\\SpatialMapping.ipynb\n",
      "Importing Jupyter notebook from D:\\Academics\\MLRepoNew\\ML2020-Uber-Data-Analysis\\Preprocessing\\DataEnrichment.ipynb\n",
      "Importing Jupyter notebook from D:\\Academics\\MLRepoNew\\ML2020-Uber-Data-Analysis\\Preprocessing\\Discretization.ipynb\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import nbimporter\n",
    "import geopandas as gpd\n",
    "import commons.utility.pickle_util as pickle\n",
    "import Preprocessing.SpatialMapping as sm\n",
    "import Preprocessing.Discretization as dis\n",
    "from sklearn.preprocessing import LabelEncoder\n",
    "from sklearn.model_selection import train_test_split\n",
    "from sklearn.preprocessing import MinMaxScaler\n",
    "import time\n",
    "from pandas.plotting import register_matplotlib_converters\n",
    "#register_matplotlib_converters()\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "coord_system = {'init': 'epsg:4326'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "mounted-issue",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "frequent-responsibility",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "from keras.models import Sequential\n",
    "from keras.layers import Dense\n",
    "from keras.layers import LSTM\n",
    "from keras.layers import Dropout\n",
    "from keras.preprocessing.sequence import TimeseriesGenerator\n",
    "import keras"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "naughty-accordance",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data for the month :apr Found.....\n",
      "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n",
      "Loading file ../data/picklerCheckpoints/spatialJoinData/uber-spatial-data-apr14.p\n",
      "Decompressing ../data/picklerCheckpoints/spatialJoinData/uber-spatial-data-apr14.p.pbz2..............Done\n",
      "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n",
      "\n",
      "Data for the month :may Found.....\n",
      "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n",
      "Loading file ../data/picklerCheckpoints/spatialJoinData/uber-spatial-data-may14.p\n",
      "Decompressing ../data/picklerCheckpoints/spatialJoinData/uber-spatial-data-may14.p.pbz2..............Done\n",
      "++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n",
      "\n"
     ]
    }
   ],
   "source": [
    "data = sm.run(['apr', 'may'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "furnished-diary",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = data.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "refined-probability",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Result Saved in ../data/encodedMappings/zone.p\n"
     ]
    }
   ],
   "source": [
    "df2 = df[['datetime', 'zone']]\n",
    "df2g = df2.groupby(['datetime', 'zone']).size().reset_index(name='counts')\n",
    "df2g = dis.encode(df2g, ['zone'])\n",
    "df2g['datetime'] = pd.to_datetime(df2g['datetime'], format='%m-%d-%Y:%H').apply(lambda x: time.mktime(x.timetuple()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "amateur-bunch",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157571, 3) (17507, 3)\n"
     ]
    }
   ],
   "source": [
    "test_size = int(len(df2g) * 0.1) # the test data will be 10% (0.1) of the entire data\n",
    "train = df2g.iloc[:-test_size,:].copy() \n",
    "# the copy() here is important, it will prevent us from getting: SettingWithCopyWarning: A value is trying to be set on a copy of a slice from a DataFrame. Try using .loc[row_index,col_indexer] = value instead\n",
    "test = df2g.iloc[-test_size:,:].copy()\n",
    "print(train.shape, test.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "according-colors",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAACw4AAAD5CAYAAAA0/oM0AAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuNCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8QVMy6AAAACXBIWXMAAAsTAAALEwEAmpwYAADE2UlEQVR4nOzdeXwcdf0/8NdAERSVr3Iogn4DiniggFZUkK+gnIIc8hMRDxQV+X69UTHcNxQKpRRo6UmhHKUXFJrebdIkTdM0d3Pf931s7k32mN8f2U2TzR6zuzPz+czs6/l48ChJdmfeOzvzmc985j3vj6KqKoiIiIiIiIiIiIiIiIiIiIiIiIiIiMjejhIdABERERERERERERERERERERERERERERmPicNEREREREREREREREREREREREREREQJgInDRERERERERERERERERERERERERERECYCJw0RERERERERERERERERERERERERERAmAicNEREREREREREREREREREREREREREQJgInDRERERERERERERERERERERERERERECWCOlhcpitIAYAiAB4BbVdW5iqJ8EsA7AJIANAC4WVXVft/r7wHwO9/r/6qq6o5wyz/ppJPUpKSk2D4BERERERERERERERERERERERERERERAQDy8vJ6VFU9OdjfNCUO+1yqqmrPtJ+TAexRVXWeoijJvp//oyjKVwDcAuCrAD4DYLeiKF9UVdUTasFJSUnIzc2NIhQiIiIiIiIiIiIiIiIiIiIiIiIiIiIKpChKY6i/HRXHcq8H8Jrv/18DcMO0369VVXVcVdV6ADUALohjPURERERERERERERERERERERERERERBQnrYnDKoCdiqLkKYpyh+93n1JVtR0AfP+e4vv9aQCap723xfc7IiIiIiIiIiIiIiIiIiIiIiIiIiIiEmSOxtddpKpqm6IopwDYpShKRZjXKkF+p8560WQC8h0A8LnPfU5jGERERERERERERERERERERERERERERBQLTYnDqqq2+f7tUhTlXQAXAOhUFOVUVVXbFUU5FUCX7+UtAD477e2nA2gLssxlAJYBwNy5c2clFhMREREREREREREREREREREREREREUXD5XKhpaUFTqdTdCiGO+6443D66afjmGOO0fyeiInDiqIcD+AoVVWHfP9/BYBHAbwP4DYA83z/bva95X0AbymKsgDAZwCcBSAnmg9CREREREREREREREREREREREREREQUrZaWFnzsYx9DUlISFEURHY5hVFVFb28vWlpacMYZZ2h+n5aKw58C8K5v480B8JaqqtsVRTkEYJ2iKL8D0ATgp75AShVFWQegDIAbwJ9UVfVE93GIiIiIiIiIiIiIiIiIiIiIiIiIiIii43Q6bZ80DACKouDEE09Ed3d3VO+LmDisqmodgHOD/L4XwA9DvOcJAE9EFQkREREREREREREREREREREREREREVGc7J407BfL5zzKgDiIiIiIiIiIiIiIhNhe0o6e4XHRYRARERERERERERFRgurt7cV5552H8847D5/+9Kdx2mmnTf08MTER9r25ubn461//amh8ESsOExEREREREREREVnBoNOFO9/IxzmnfRxb/nKx6HCIiIiIiIiIiIiIKAGdeOKJKCwsBAA8/PDD+OhHP4p//etfU393u92YMyd4+u7cuXMxd+5cQ+NjxWEiIiIiIiIiIiKyBY9HBQC09I8JjoSIiIiIiIiIiIiI6Ijf/OY3uOuuu3DppZfiP//5D3JycnDhhRfi/PPPx4UXXojKykoAQFpaGq699loAk0nHt99+Oy655BKceeaZWLRokS6xsOIwERERERERERERERERERERERERERGRgaqqqrB7924cffTRGBwcRHp6OubMmYPdu3fj3nvvxcaNG2e9p6KiAqmpqRgaGsLZZ5+N//3f/8UxxxwTVxxMHCYiIiIiIiIiIiIiIiIiIiIiIiIiItt55INSlLUN6rrMr3zm43jox1+N+n0//elPcfTRRwMABgYGcNttt6G6uhqKosDlcgV9zzXXXINjjz0Wxx57LE455RR0dnbi9NNPjyv+o+J6NxEREREREREREZFgHq+KxWk1GB53iw6FiIiIiIiIiIiIiCio448/fur/H3jgAVx66aUoKSnBBx98AKfTGfQ9xx577NT/H3300XC74x8HZ8VhIiIiIiIiIiIisrTtJR14ZnslSlv1rRpBRERERERERERERNYWS2VgMwwMDOC0004DAKxevdrUdbPiMBEREREREREREVnauNsDABiZmKy0oKoioyEiIiIiIiIiIiIiCu/uu+/GPffcg4suuggej8fUdSuqBKPoc+fOVXNzc0WHQURERERERERERBa0Kb8Fd60rwqVnn4zUym6c8OFjUPTQFaLDIiIiIiIiIiIiIiIBysvL8eUvf1l0GKYJ9nkVRclTVXVusNez4jAREREREREREREREREREREREREREVECYOIwERERERERERER2YL4udWIiIiIiIiIiIiIiOTGxGEiIiIiIiIiIiKyNEURHQERERERERERERERkTUwcZiIiIiIiIiIiIiIiIiIiIiIiIiIiCgBMHGYiIiIiIiIiIiIbEVVVdEhEBERERERERERERFJiYnDREREREREREREZAuK6ACIiIiIiIiIiIiIiCTHxGEiIiIiIiIiIiKyBdYZJiIiIiIiIiIiIiIZOBwOLF68OKb3Lly4EKOjozpHdAQTh4mIiIiIiIiIiMjSFNYaJiIiIiIiIiIiIiKJyJw4PMewJRMRERERERERERERERERERERERERESWY5ORk1NbW4rzzzsPll1+OU045BevWrcP4+DhuvPFGPPLIIxgZGcHNN9+MlpYWeDwePPDAA+js7ERbWxsuvfRSnHTSSUhNTdU9NiYOExERERERERERERERERERERERERER6WTevHkoKSlBYWEhdu7ciQ0bNiAnJweqquK6665Deno6uru78ZnPfAYpKSkAgIGBAZxwwglYsGABUlNTcdJJJxkSGxOHiYiIiIiIiIiIyFZU0QEQERERERERERERkRy2JQMdh/Vd5qe/Blw9T/PLd+7ciZ07d+L8888HAAwPD6O6uhoXX3wx/vWvf+E///kPrr32Wlx88cX6xhkCE4eJiIiIiIiIiIiIiIiIiIiIiIiIiIgMoKoq7rnnHvzxj3+c9be8vDxs3boV99xzD6644go8+OCDhsfDxGEiIiIiIiIiIiKyNEURHQERERERERERERERSSmKysB6+tjHPoahoSEAwJVXXokHHngAv/jFL/DRj34Ura2tOOaYY+B2u/HJT34Sv/zlL/HRj34Uq1evnvHek046yZDYmDhMRERERERERERERERERERERERERESkkxNPPBEXXXQRzjnnHFx99dW49dZb8d3vfhcA8NGPfhRvvPEGampq8O9//xtHHXUUjjnmGCxZsgQAcMcdd+Dqq6/GqaeeitTUVN1jU1RV1X2h0Zo7d66am5srOgwiIiIiIiIiIiKyoM2Frfjb2kJccvbJSKvsnvr9sl99E1d89dMCIyMiIiIiIiIiIiIis5WXl+PLX/6y6DBME+zzKoqSp6rq3GCvP8qUqIiIiIiIiIiIiIhMti63RXQIRERERERERERERERSYeIwERERERERERERWZZjdEJ0CEREREREREREREREljFHdABEREREREREREREsShtG8A1izLx/S+eLDoUIiIiIiIiIiIiIiJLYMVhIiKyhOy6XhQ2O/CtJ3bjlX21osMhIiIiIiIiCVR2DAEAMmt6BEdCRERERERERERERDJRVVV0CKaI5XOy4jAREVnCLcuyp/5/3rYK3Pn9zwuMhoiIiIiIiIiIiIiIiIiIiIiIZHTccceht7cXJ554IhRFER2OYVRVRW9vL4477rio3sfEYSIiIiIiIiIiIrKpxKgoQURERERERERERERHnH766WhpaUF3d7foUAx33HHH4fTTT4/qPUwcJiIiIiIiIiIiIltIkJnniIiIiIiIiIiIiCiMY445BmeccYboMKR1lOgAiIiIiIiIiIiIiOJh34nmiIiIiIiIiIiIiIj0pTlxWFGUoxVFKVAUZYvv508qirJLUZRq37+fmPbaexRFqVEUpVJRlCuNCJyIiIiIiIiIiIhoOoUZxEREREREREREREREYUVTcfhvAMqn/ZwMYI+qqmcB2OP7GYqifAXALQC+CuAqAIsVRTlan3CJiIiIiIiIiIiIiIiIiIiIiIiIiIgoFpoShxVFOR3ANQBWTPv19QBe8/3/awBumPb7taqqjquqWg+gBsAFukRLREREREREREREREREREREREREREREMdFacXghgLsBeKf97lOqqrYDgO/fU3y/Pw1A87TXtfh+R0RERERERERERGQYVQ3/MxERERERERERERFRoouYOKwoyrUAulRVzdO4TCXI72YN0SuKcoeiKLmKouR2d3drXDQRERERERERERHRTEqwEUkiIiIiIiIiIiIiIppFS8XhiwBcpyhKA4C1AH6gKMobADoVRTkVAHz/dvle3wLgs9PefzqAtsCFqqq6TFXVuaqqzj355JPj+AhEREREREREREREREREREREREREREQUScTEYVVV71FV9XRVVZMA3AJgr6qqvwTwPoDbfC+7DcBm3/+/D+AWRVGOVRTlDABnAcjRPXIiIiIiIiIiIiIiAKpvvjNWHiYiIiIiIiIiIiIiCm9OHO+dB2Cdoii/A9AE4KcAoKpqqaIo6wCUAXAD+JOqqp64IyUiIiIiIiIiIiKahonCRERERERERERERETRiSpxWFXVNABpvv/vBfDDEK97AsATccZGREREREREREREpJm/8vDUz2LCICIiIiIiIiIiIiKS1lGiAyAiIiIiIiIiIiKKBysPExERERERERERERFpw8RhIiIiIiIiIiIiIiIiIiIiIiIiIiKiBMDEYSIiIiIiIiIiIrKUV/fXIyk5BeMur+hQiIiIiIiIiIiIiIgshYnDRERERERERERENENt9zCa+0ZFhxHSy6k1AIAhp1twJERERERERERERERE1jJHdABEREREREREREQklx8+tw8A0DDvGsGRaKNAAaCKDoOISCi3x4sJjxcf+RBv/RARERERERERUWisOExERERERERERES2pKpMJiaixPHXtQX4yoM7RIdBRERERERERESSY+IwERERERERERERERGRxW093CE6BCIiitO428OH34iIiIiIyHBMHCYiIiIiIiIiIiLpjYy7Ud8zIjoMIiIiIiJDdAw4cfb927Emu1F0KEREREREZHNMHCYiIiIiIiIiIiJT/OH1XDy7ozKm9962KgeXPpumb0BERERERJJo6hsFAHxQ1CY4EiKSyZK0WvxiRbboMIiIiMhmmDhMRJaVVtmFzYWtosMgIiIiIiIiIo12lXXipdSamN6b29gPAGjsHUHP8AQAQFFmvoaTOhMREREREZGdPL29AvtrekWHQURERDbDxGEisqzfvHoIf1tbKDoMIiIiIiIiIjLRb189JDoEIiIiIiJDuTxeXP9SJjKre0SHQkRERERENsTEYSIiIiIiIiIiIrKMcbdX82tZgZiIiIiIrEZVgc5BJ4paBvCfjcWiwyGypZLWAazLbRYdBhEREZEwc0QHQEQUi+IWh+gQiIiIiIiIiIiIiIiIiHShKKIjIEoc176YCQC4ee5nBUdCREREJAYrDhORJV330n7RIRARERERERGRYCpLChMRERERERERERERRYWJw0REREREREREREREREREZCtujxd/X1uA6s4h0aFEhc/GERERERGR0Zg4TEREREREREREREHtKO1AXfew6DA04+zORERERORX1j6I9wrbcNe6ItGhaMK+LBERERERmWWO6ACIiIiIiIiIiIhITn9ckwcAaJh3jeBIYqOyXBsRERERWZjKDi0RERERERmAFYeJiIiIiIiIiIjIkhSWZSMiIiIiG1LY0SUiIiIiIgMxcZiIiIiIiIiIiIgsbcLjBQAEq8dW0zXMSm1EREREZBnsuxIRERERkdGYOExERERERERERES2VNziwGUL9mFFRr3oUMgk/SMT+N7Te1HePig6FCJhWh1jyKnvEx0GERFFiUWGiYiIaJb+RqApW3QURGRDTBwmIiIiIiIiIiIiW+ofdQEACpsdYgMh06RXd6OlfwyL02pFh0IkzPefScXNSw+IDoOIiIiIiIji9cLXgVVXio6CiGyIicNEREREREREREREREQ24fZyinsiIitjK05EREREREZj4jAREREREREREREREREREZFQyqzfMImYiIiIiIiMwMRhIiIiIiIiIiKLKWjqx1/fLoCXFQUpwalq4M88JoiIiIjI+manEBMRERFJbLAN6K4ECt8C0p4WHQ0RaTBHdABERERERERERBSdP7yei57hCTxw7Vdw8seOFR0OERERCeT2eOHmw0REREREREQkyoIvz/z5kv+IiYOINGPiMBEREREREREREdlCRnVP0N+rnOSZiGzs16tykFXbKzoMIiLSCSfRILK2l1NrcP5n/wsXfuEk0aEQERERhXSU6ACIiIiIiIiIiMh63B6v6BCINGPyBRHZGZOGrauw2YHP37sVXYNO0aEQkQQURXQERKSH+TsqceuKg6LDICIiIgqLicNERERERERERBSVrNoefOG+bchr7BMdCiUgJlSQFiqzxYnIAl7dXw+PV2XyN5FB2B0gIiIiIiIKjonDREREREREREQWpULMnfCM6h4AQHYdE4dJrIP1TLSimRRmlhMlpM2Frbj6hQzRYRCRpKzcPZie/Ly/pgdtjjFxwRARERFpNdoHVG4THQURhcHEYSIikto1izIw9/HdosMgIiIiIosbGXfjqa3lGHd7RIeiEwvf+SZLauodNb2Ca059H9blNod9ze7yLpOiISJKTD3D4xhyukSHEdHf1haivH1QdBgRdQ46MTZhl/4okXVtzGtBUnIKugadokMJSkXwZOdfrDiIK59PNz0eIpJDVm0PNuW3iA6DiEibtbcCb98CjPChfyJZMXGYiIikVto2iJ7hcdFhEBEREZHFvZRag6XpdXgzu0l0KESW9D/zU7Fqf4Op67x56QHcvaEYzX2jKG5xTP2+pZ9V1oiIzDL38d24ZH6a6DBs49tP7sEvVmSLDoMo4b3jezitrmdEaBwZ1d0YnPZwhpbHQ4fG3cYFRERSu3X5Qdy1rkh0GEQkWso/gf0viI4isr66yX+98j+ISpSomDhMRERERERERLbncnsBAB6vuRVTiewkv7FfyHovfiYV1720X8i6iYgI6B2ZEB2CreQ3OfDnt/Lx9Yd3iA6FKGH0j07g5dQa02fQCKd3eBy/WpmDP72ZLzoUooTmdHkwMMqkNiKykEMrgF0Pio6CiGwgYuKwoijHKYqSoyhKkaIopYqiPOL7/ScVRdmlKEq1799PTHvPPYqi1CiKUqkoypVGfgAiIiIiIiIiooQlz31vIqlJlCNCRGR5424Psmp7RIdhaVuK2zHoPFI1VGWnjshQzX1jmL+jEgdq5Zkqe9z3cGtN1/DsP7LzSmSaH7+YiXMf3Sk6DCIi+5K9X5P9CpA+X3QUREJoqTg8DuAHqqqeC+A8AFcpivIdAMkA9qiqehaAPb6foSjKVwDcAuCrAK4CsFhRlKMNiJ2IEtDohBsuj1d0GERERLq5c00ebniZFfSIiCg6ipY5bIkkNvfxXXgipUx0GEREFIPHtpTh1uUHUdExKDqUWS6atxfbS9qRlJxiieqB7NIRmcsl4Qw0w+NuJCWnYHNhKxRe6BGZrjpY8j4REenAIv2a7f8B9j4uOgoiISImDquT/L2lY3z/qQCuB/Ca7/evAbjB9//XA1irquq4qqr1AGoAXKBn0ESUuL7y4A789tVDosMgIiLSzfbSDhQ2O0SHQURERGSqnuEJLM+oN219rORIRKSfqs7JW0YOCRNzWx1jWLKvDgBQ18NEICKS35Cv+vji1FrBkRAlhr0VnXhhd7XoMOKyKrMemwtbRYdBREREFjdHy4t8FYPzAHwBwMuqqh5UFOVTqqq2A4Cqqu2Kopzie/lpALKnvb3F9zsiIl1k1nAaPCIisp/fv5aLjx57NBbecr7oUIiIiCKySL0I0hu/eLIQpooT6WvC7cWcoxQcdRRPBkQUXE3XEJr7xnDpl06J/GJB1ICpsmWfORsAvKqKCbcXH5qjZSJhIork9tW5okOI26NbJmfvuf48puEQ2ZqjCWgrEB0FEdmYpisMVVU9qqqeB+B0ABcoinJOmJcHGzWaddmlKModiqLkKoqS293drSlYIiIiIiK72l3eifcK20SHQUQUtfSqbryW1SA6jIQj6ga3Y3QCLf2jYlZOwg2MunDPpsNwujyiQyEKiSmNRMb44v3b8OD7JaLDCKpvZEJ0CERSq+8ZweNbymYlzertsgXp+O1qa8wYKXt/QcWR6u5dQ+P44v3b4PZ4xQZFRKZLrejCmuxG0WEQkShLvw+s+7XoKHRggSe1iBKUporDfqqqOhRFSQNwFYBORVFO9VUbPhVAl+9lLQA+O+1tpwOYlQGhquoyAMsAYO7cuWwliIiIiIiIiCzo16tyAAC3XZgkNhAyxfeeTsXwuBv/d8nnRYdCAvhnADrntI/jF9/+76jeW9ziwFGK7CkaREQUzhvZTXj8hq+JDmOWuY/vEh1C3KxQ9ZSs6w+v56Kmaxg///bn8PmTPyo6HApjd1knAKC5bxS3+a61/dxeNhREicYqD2MQkUHG+kRHEB+OAxJJL2LFYUVRTlYU5b98//9hAJcBqADwPoDbfC+7DcBm3/+/D+AWRVGOVRTlDABnAZh5ZUOU4LxeFQ09I6LDICIiIiIiSjiqTSociBp3HR53i1kxWd51L+3HtS9mClu/DAlZSckpeHCznNU6iYhiIUPbCgCBuXStFpodQeHNdJpmyOnCY1vKdJ/dwcuE0ykyb4mRCTdeSq0BAPSPugRHQ6QPp8uDx7aUYcjJfZqIiIhINhEThwGcCiBVUZRiAIcA7FJVdQuAeQAuVxSlGsDlvp+hqmopgHUAygBsB/AnVVU5fyHRNIvTanDJs2mo7BgSHQoREREREZEtNfeN4pL5qegcdAI4kmh7qKEfVy1M1/1mPBGRFq8f4DSzRGRtly3Yh9LWAdFhhNUzPAGASblkPYv2VGNlZj3eOdQsOhQyQWATxQRvsqN1uc1YmVmPRXuqRYdCRERG6aoAOsI8KF/+AbDqKnmePCWiKXMivUBV1WIA5wf5fS+AH4Z4zxMAnog7OiKbOtTQDwBoGxjD2Z/+mOBoiIiIiIiI7GdNdiMaekfxXkEr/vj9z0/9fpdv6tearmGcc9oJosKznKyaHgw6WW2YJnGcn4gocdV0DYsOgci2XJ7JTpaHCaS6sdKWtFKsRFr52zW3JO3aPZuKRYdARGQ/i78d4g++p6S23W1aKEQUnYiJw0REREREREREWj3wXgm+fvoJ+Oncz4oOZQa7Jjqa9bFuXXEw+PrtumET2FNby3H6Jz4sOgzdcU8lIko87KeQVj3D42juG8X5n/uE6FDIaGrgj2wniBLJ2zms6E5ERETkd5ToAIgSmkXGI1KK21HaJvf0c0RERHbzbkELarqGRIdBRBS1NdmN+PcGeSq4jEx4sCStFh6JE0fK2wfxQVGbpte+ebARSckpmHB7DY4qPM78bV9L0+vwwOZS0WEQxU/eZp+IACzYWYmk5BTRYRABAK57MRM3Ls4SHQaZSOT1jNer4pV9tRgZ54wuZH98iIem21LchrK2QdFhEBERkQ8rDhMJYLUbrH96Kx8A0DDvGsGREMWvpmsYL+ypxoKbz8UxR/P5GRGe31WFsz/9Mfzoa6eKDoVIav94pwgAz79ERPFatKcaAPCZE44THEloV7+QAQD48bmfifjaF3ZPfp6BMZehMRERWZXVxt2IEtWivTWGr0OxUIPAvCqx2gacokMwBfczoLZ7GJsKWuH2iNsYO8s6MW9bBbLreoXFQGQUr1fFvzcU47cXJeGc006Y+r0C65yTyTh/fqsAAO95ECUMC12PESUqZkwRkSncHq/wilhEAPDvDUX4oKgNxS2soi3KC3uq8X9v5mPc7YHbw3aBKJKNeS3oHEyMG1hEREaaYL+DSDdWzDlhogxRYvjX+iLc/MoB0WEkDNWSZwS58FY6mYI72pTHU8rxQVEbSgTOsjnu9gAAKw6TLXUMOrExvwV/eD1XdChRO1jXi7zGPtFhEBEREZmGicNEAiXSwOq1L2bii/dvEx0GEUnk7Pu34+alvJlHFMk/1xfhtlU5osMgIiIiG3vzYCO2l7SLDoOIKG4b8lqQ08CEj0SSVdMjOgQiIl0wv5vszApFJ3+2LBs3LeE9KyKSWNFaoHid6CiIyEaYOEwkgAWujQAARc0OFDU7dFlWRceQLsshInvJb3KIDoHIErqGxkWHQEREkmIVVdLDfe+W4M438kWHkTC++dgu7CztEB2GJnmN/fjKg9vRNzIhOhQioqBuXXFQdAhERDFRAu4WenltRzbEMQsiIh29+0dg0x9ER0FENsLEYSIK6fqX9+P6l/eLDoOIiIiIiIiIyDZ6RybweEq56DA0WZJWi9EJD3JZwZWILMgqBTyIAjHPTh81XUNQI2UtCtzYgbOSdgw6BUVCZDyek4kooXRXAmnzREchgYDWn0+TEEmHicNERCZYkVGHP67JFR2GrWw73I6xCU8cS2DHlIiIiMjOZt+UssdtKtmm9+R4b4Ky5BdvxZiJyE4m3F5sKW6LnMRGuuGWJquRra9vZbkNfbhsQTpeP9AY9O+B1X7NpPCLJiIisrdXfwSkPQU4B0RHQkQUFhOHiQSy2hhxUnIKFqfViA7Dkh5PKceO0k7RYdjK/76Zj4ffL436fYkyJNcx4MSE2ys6jJi09I+KDoGIiIjIFCPjbvSNTMS1DFH3nEXeaKfgsmp68EFRm+gwiIgohOd2VuLPbxXg16tykJScgtEJt+iQwnqvoBXZdb3hX2Sx8W2ZcVNSMB6vijbHmI5LlH9P++e6IiQlp8S9nIbeyTHm4hZH3Msi7fIa+7E+t1l0GCQRf2VtK+XLJyWn4K9vF4gOg4iszBPfeC8RkVmYOEwkgOxPEzf2jqCiYzDo31Zk1JscDVFobQN6Dprah9PlwXee2oPkTcWiQ4laakUXvvd0KraXtIsOhUiYwy0DOt8UIiK7WL2/Hg9tLhEdhmXIf0scuOL5dHzjsV1RvScwYddqD6SScW5dcRB/4c3NkGI5VvKb+tE1lJhTRld0DOKGl/djzCV3YmM4gdN/E4nWNjDZnmRU9wAAHKMuw9fZNehEQVN/TO/9+zuFuGVZts4R0Sxy3yogwZ7dWYkL5+1Fe5zj4EbtZuNuD1Iru3Rd5sb8Fl2Xx96AuW5akoV/b7DefQkyjv86TPZ744He50O5RETxs1jbT5SImDhMRLN8f34arlqYIToMzZwuj+gQiKTiPyb2lOs7aGuGktYB37/BH17Qm6qqSEpOwcuprKZO8vjxS5m4cN5e0WEQkYQe/qAMr4WYZpWsqZUPipBFJUoCxk8WZ+FHLxg7PtLSP4qk5BTsq+o2dD3ReiKlHIXNDhxqiC3hUCRWRCc64oqF6bhxcZboMIRha0BWl1E92T/oGZKzat0TKeX47auHUNDUH/dMKnqbOv5DdFz1eMCouW+yH5fpeyCEiI6Q7SHnezYV45yHdogOg4iIiEgqTBwmEki2iyarumphuugQiMji5u+oFB0CUVgqOw1ERLpj00pEWvQMG5sEk9/kAADpp3SWucl0ujzIqe8THQaRlMyoakxEiau+ZwQA8MDmkqhnUglU1z2Mlv5RPcKCy+PFP9cXaXptPH2cQw2T/Q+9qyQTWZmsxSXfzmnG8Lh1Z1MhY+h57iEKigPQRCQ5Jg4TCSDpNZNlNfSyQ0/RYz+diIiIrGpvRSdK2wZEh0FERHFwebxISk7BO4eaRIcSlhXGsB75oAw3Lz2Amq5hAPpUEKTZBsZceC2rgQ81xiHU8dQ9NI63c+RoCy5+Zi8e21ImOgwishg9Zo/7wXP78L2nU3WIBqhoH5r6/1BnLZEzFFihf0UULxUqXj/QgIExsQ8xebwqxiY4cy0Fp+e5h2gmnu0ncTsQyY6Jw0RElFAUWR931hHv4REREdnb7atzcc2iTNFhkCAJ0J0lSgj+G+j7a3oFR2J9FR2TyUqBSQkiE4JkMeh0obpzKPILNbh302E89H4pchv7dVkeHXHHmlzcs+mwFNXOmvvGsDKzXvPrOQSlHyblUzh8KCZ2PLaIxOgcHMeDm0uxOK0WgLjUsb+tLcCXH9wuaO1AWdsgRicSu9Kx2+PFPZuK0dwnvq9LFJTXCzQfEh1FYugsAbb9h8kMRBJh4jCRQDwdRuf21aE7bF6vPbZm15ATZ96TgoKmfrQ5xrC9pANuj1d0WNJKhCTgeFhp8/SNTOAL925FTgOnlw2ltG0A2XVMKiAiInk09Ixgb0VnyL+7PV6pb9Jaqa8UjugkAhXAuwUtcIxOCI2DzKXnoW1WW7Gnogur92tPhiOyg58vy8blz6frsqx+Xzs/4ZZznOrl1BpctTD4Zx1yunD2/duQVtllclTh+Vu+3uHJbeuxyfgmRYcPOVAw/r6R3vuHxJdnhknAj0xE02wpbhe6/h8tysBf3y4QGoNouY39eDunGf9cXyQ6FKLgDr4CrLwMqN0rOhL7e/36ye090i06kkmv3zCZyEyUwJg4TGQip8uD+987jCGndZ8sFHnjf29F6MH9dwtaTYzEOFk1vfCqwOqsBtzw8n7c+UYevnDftqnqOUR2ld/YD7dXRUZ1DwD7JPLo6ZpFmbhlWbboMMjm+kcm0DHgnPq5uW8UQ06x08kRkbwueTYNt6/ODfn3L9y3DcvS60yMaKbA7oRd+heyfAz/9mzoGcE/3inCXxL8RhjF7gv3bcO87RWmrOvhD8pMWQ+RLErbEmc8af6OSlR0BK+uXNU5jHG3Fy/sqTY5qpns0hchIv2pqory9sGQicLD4+64KjUaVYCDhT1ID+0DY+gf4YOotpfAzQVn7CCSXJdvrMjRLDaOWGQsAFryREdhPfueAdoKgbrUyURmogTGxGEiE63LbcYb2U2sqGmABzaXiA5Bd11D41P/X9wyIDASImNdtTAdHxS3iQ6DiAB84/Fd+M5Te6Z+vviZVFz/0n6BERGR1W3MbxEdAhls3Fd5smtwPMIrgbU5TbhxMc8rNNtb2U2iQyCf3uFx7CkPXU1ebqwpSNaWiJVAKTaqquLdgha4OFOdbazPbcHVL2QgrSp48ZRblx/Exc+kmhwVGS3adv+dQxZMaNLgu0/txQVP7hYdBhmMFfaJKDEZfJG35xFgxQ+MXUc8ZG36U58Aln1fdBREUmDiMJGJvAHTzck8ba/VjE54RIdAAsTT19Tj6PN6VTT1xl7pwShWa1kqOoawuZCJw0QyCNY1qesZMT8QIrIkx+gEHKOsEhSP7LperM+1581gAEjedBgFTQ7RYZCEJjxePLm1PKFnOpClYt+vV+Xgd6/lYsw3zuIPS+YhrFBbTrXc1THZXeC+uimvBUnJKegacgZ9PdnHkNOFpOQUrMjQZzaOD4rb8Y93irAkrVaX5ZF4Ze2TFeLrA8ZgJOkeWM707RapDyOijxPr9/rQ+6VT///k1nJMuO3z8IDLw34bERGJYsA5iH24EHi+J5INE4eJiMgUT24tx/O7qkSHoWs//eXUGvzP/FTUdA3ruFT92PWaZEVGHe7ZdFh0GERERBTgvEd34bxHd4kOQzMZk+BuWZaNf28oDvuawORCGT8H2Z/eD0KPu71Yll6H53aKv2ZMdA2+hCWPhRsXVjOjUGTbrdfnTc7M4HTZJ/GLgusZnny47o3sRl2W539Yr3so8owPRGRPy9LrbDfDT2kbZ94kilZ+Uz/aHGOiw9BGsr440Sx8YstE3NZEsmDiMJGJ7NAflqUCTjhXLUy3/PS3st3I0MOy9Dq8sKdadBi6OljfBwBoH7DIRbmFhGtpHk8px9s5nMqYEosNTwsxy6nvw/aS9qmfNxe2orDZIS4gIgqLyVsUzBhnjKEQOOW6PNh6k53IOpwZWBVbpuu+4XG3ptfJPoYp63cfjOSbkkwi+zFlRTLPQBBvG+X2yvvZYvHM9krRIUivpX8USckpSK3oEh1K1Kx0ThbN6fJontHrJ4uz8L2n9xocUXz41ZP0ZDydZi4EHj5BdBQ6CD1HFJmk+RBweIPoKEhiTBwmItup6Biy7PS3IS+c2XcKigMNRESJ5+alB3DnG/lTP/9tbSFueNnaDwwRkTESpQstuk8cy434q15INyASikVd9zCaekdFh5GQ7JoYVNDUj4FRl+gwyEB23XfNYIWCDH7nPLRDdAgRtTrGUN05JDqMuFlotzDNhNuLrJoe0WFE5HR58KuVB1Glw35o1H4w7vbg16tyDJsxzwq7b3Nf8KIfMicUa8aTsvRUVcW+qm7dZmzx3/vcYMFq01ZoL4wS7dd/89IDUc3oZbNnCIgE8B9EErVU+54WHYE2Gc8BWS+GeUFgAxXHNva4gdrU2N9vJ7WpwIbbtb125WXAxt8ZGw9ZGhOHiQRiP54oPhJ136Wi97TBRJR4nC4PnC5WgyQiotDi6Ys3MlFVGj94bh/+Z/6RQfc/vZUf5tWUiKK5vFRVFTcuzsKvVh00LqCw6xey2oRhh+RG7iL2ctG8vbj8eT6MZEdPbSvHrSsOorjFITqUsPIa+5FR3YOH3y81bB3xNr1FzQNIr+qe+pnt4Gwiz2/su9jfWzlNuG1VDt4rbBUdCllIccuA6BCisi63GUnJKbyfQNZnRKcglpN9UzbgssjY6Z5HgZ33a3/9WF/s69o3D1hzA1CfEfsy7GLNDUDJRtFRkE0wcZiIiBLSsNONsrZB0WEYykqVbIhILhc8sRtfemC76DDCWp/bjNcPNIgOg0io/2woRnl7/P2ZvMZ+PPx+KR8+IiKkFLdreh1bC/3JfvUWzeWl2Te6ee1LkVhlD2FfzP74DWvjr47bn0AV7BXLtFSkB//3HW+bUNs9En8wZKiW/smK120Op74L5gklIby4p1p0CJq8sHsyzp7h8bCvs0WldyLN4ujbFb6pXxjSimH79NZM/jvSpW8oRAmOicNEFBUOYJtjwu0VHYJt+e8n/nb1IfxoEZ9IIyIKZtDpFh1CRP/eUIwHNxtX1YfICt7JbcYf1+TFvZyblmRhdVZD/AEFEU0ul6qqGHfrV50kcNV2uRUfuE15iUbztlXg6hd4bSOz7SXtaO6zSLUYn8CmhW0NUfwi9UW8KscE7cqofigTcIjiZ+U+jlHX8SQvKz8vZ+XYRXluV5XoEHTBBz1JelbuDMSraifQVWHgCkIc/57wDxoQkXmYOEwkUCL3QSi87aUdM37mIHBieHFPNQqa+uNeji32Fg4kxGx43I17NhVjZFz+xE8iIqLpFu6uxtn3b8eQM3EqixHp4ZV9tbpUHyfj3PlGPq5amC46DE0Cb+pa+dKM424ku8B99M9v5eOL928TE0yMOGapoyg2pYWbZopCSau1pqqXRWXHEFyeyA9hsMqzfKzc76RJTb2jGBwz7r5Efc8Ihnnfg4gMk4Anord+Ciz+tvnrff6rQPVu89dLRLMwcZiIppS2DaCw2SE6DFv505v5eHxLmegwbEuWp1T1uhn53K4q3Lg4K+7l+OORY+vExsqxi7ZsXy3ezmnGqsx60aEQzVDSOoCvP7wj4pRlRJS4NuS1AAAGxsxJHGaiizZZtT2o7+EUuDQTEzKjNzIRvqK6JJe3U6z8Hcu2LUkiMe7YNV3DyK7r1TmY2So6hgxfB8mHTVZoes5+uGhPNX618qBuy9Nb4LlrY34rrn0xE0UtcicPy3bO7Rx04sqF6Xjo/cgzVOl5PZjIM3X+6IUMvJHdKDqMhCPreMb/zE/FtS8Gnw1Hj2T9S59Nw63Ls+NejtkSpY3IqO4RHQJRjBLjGJVOXaroCIgIwBzRARBZSVPvKD59wnH40Bx75txfsyhTdAi2k3K4XXQItiZ6XFK2gdFAssdHxvDy+taW7DC2uDyjDoNONzKre3DD+aeJDoeIiDS6dbm8CRZkX3euyZs1Ew+ZI/CmtpWq8QV2mb2qinnbKvCbC5Pw6ROOExITySHeB88vW7APANAw75o4A4nv7WR9el3b22CIICwjikUsMHC6dyPGbKo69XmQIHBT2r0Z8j+EWmxwwrXHq+Lp7RU45WPHxvR+O43bl7UP4v73SvDL7/y36FCkpPdXbVTfvK57GEknHo+jjop/+S5P8EZRr/3e6OM7XknJKbj266eKDkOIezYdxs8v+JzoMIhiZ6cTtCyM2KZ2uGFKJBF7Zj8SGWBg1IX/mZ+KB94r0XGpPKkRWRn7pURERCTa6IQHu8o6dVlWovVtOgfHTakeqLfA8VbRX1ui7TdkDtmShu9ck4c1BxpEh2Eqf1ujpZqZ2+PFdS9lIq2y2+Coggt1Gyq/yYFX9tXirnWFZoZje7JWuCOSmVE5CFZ6uMOO9Nz6Zn2TsbTgI+NuXL5gH2ernCa9qhvL0uvweEq56FCI4lbePogfPLcPS/bVig7FNrYUs6hUMBw/ImkZsXPGdQGQIH38qLd7nNtl67+BvY/Htwyzlb0PuDmLKxmLicNEGg2NTz6pnFnDaTaIrCxwQF2PKYJkegCxa9AJt9dryLIHnS4MOc2ZOpyI9NMzPI5xd/ipsYnIunqGx/GH13NRGcfU1jL1Zcz0h9dzccsy86e4HJ1wwzE6EdN7D7cMoLlvTOeIYpSoOw4lpO2lHXhgc+Qpr6MRmHwp2xEVTTz9oy4UtwzgX+uLDIsnFv7rfZfHmGvkRGOHBEVRuQo7SzuQlJyCNoe2c3hScgo6BpwGR5UY7LDfEolW1OJAddcw5m2zV5JsPO2Dh1O+GYIttnZ65ri19k/2T/Ib+/VbKNE0HD4i6+DOams5y4D0+aKj0K4uDVj3K2DPo6IjIZtj4jCRUNbrfBgxTRgdEWr78ilMe1u0pxovp9bEvZyRcTcueHLPtMro+h6vX394J7728E5dlxkKmxoi/cx9fDd+tzpXdBhEZLDhcbfoEIKK5fphbMKDLBMf2Gx1jOEni/ejfyS2ZN5oXbkwHec9uium9/74pUydo4mD7yIlmqqkZB/8tu2psNmBkYlQD5zxWydjpBS34653CkWHYagNeS0AgNLWQc3vKe/Q/lqROGYZP27C0PTYNi39oyhvt8bxZEVGDuG+kd2Ix7aUGbiGSbyO0ceCnZVYls6KtaHoUcAG4H0TsrZcJqcTJaBwJy5BfbAt/xCz3miN+drMgWaxcZDtMXGYSCj9ToaO0Ql4TXjKefrFrdPlweiEnAkKfk6XByOSJlFQ/KIZJOkcdOK1rAbDYonHgl1VmL+jMu7ljPpu8O4u74p7WURkL/HOmLA8vU63hDrekCGiSJI3HcatKw6iqXfUlPW9klaL/CYHPihuM2V9/orB1yzKMGV9RtNaqev3rx2a8fMPnk0zIBqyujcPNuHLD2wXHUbC+ve0isH+HpuVqnWqKkx7CIT086e38rGpoNWUdUW7N9d0DeHdghZDYklkT2+vwOfv3WrKuux+/WnXz6fnmed7T6fi6hfk73czGXC2+98rwcrM+qjeE81mlGGb65VMKoNFe2vw5NYK0WFIx6j9TKZd57xHd+L+9w5HfN3ohAeL02piqtidUtyOktaBWMIjk024vZpmLi1tG8AW3zhc38iErdpDMtkHfwOe+XycC/Htf0Y02r21QMEb+i+XYpe7SnQERFJh4jCRiYzq8/aPTOC8R3fh2Z3xJx5G46J5e/GVB3eYsq6aruGIr0mv6p71u6sWpuOrD5kTI8ntjjV5eOj9UjT365+AwutZIrK7J7aW4+6NxXEtQ4L7MUQ2F3+HRJYuTVXnEAB5qyhPuL3405v5qO2OfI0STmlbYlU+C3y4ra5nRFAkJLsxV6iKt/bC60j9+O/tbT3cjvMf24U8VrIinVy2IB3/eKco8gtjlKjtwJK02piShuwgWKJvLNfKvL6Wi10TuMk4PIYnuT1e5NT3iQ6DfGJtyRyjLryR3RTxdSsz6/HM9kpsL+mIeh1/eisf175o/gxMf19bgDIdxm4S6Szxy5UHNc1ces2iTPz5rQJUdAziG4/twjuHWFGTYpS3GhiNc+Y69cij07pbeRmw+U/RvUeGp5uIKGFETBxWFOWziqKkKopSrihKqaIof/P9/pOKouxSFKXa9+8npr3nHkVRahRFqVQU5UojPwARAX2jk9VUYrnYitXO0g70mljF5bIF+yK+5tercmb9rsGkKmVkDJfHG+EV2jvOA77jRM8bE6H67Qdqe5HfxBuVRGQvWioFiOR0eUyZfYFIhKrOIewu6zRs+UYORYoY5qzpGkJScgoqfQnIU7HoFEx+Uz9SDrfjnk2RK+oQEVlFc98oqqc9tG3FhMbsul4Ak9WrAGt+Br2syKjDVx9k9e5AonaJqfXy/q/pZKmarmcceY39OOBr70gwOXavsCwQom7GJjwJlZznd8Y9KdiQZ83q+At3V+PmpQemfp7weOGOeE+IAH1zyvRa1rhb2wOgWl8ng/cK2/Dnt/NFh2GKeMbWp+9DwR4GCLeLVXdOXoNmxDljIpF8EqkXFqNoB20SNaFaVQGXE/AGOX8m8sAX6UZLxWE3gH+qqvplAN8B8CdFUb4CIBnAHlVVzwKwx/czfH+7BcBXAVwFYLGiKEcbETzZV+/wOP6zoRhOHarMtDrG0DXk1CEqmu6ONXmiQ0goiXrK/5EJ08gZ0Z/6+fJs/GRxlv4LjpKV+89m3dixY3+aVU4oUX3pge14dEuZ6DCIDHHF8+n4/eu5osOwjK2HJx+oTKucOSNJpP6FHfsFREbhNKL6UwRfwP34peAVvEKFtSKjbtbMT7LsFbIkCor0eEo5RiaMScoYnfCgsmMo8gslwj0icd29sRijE3LOYhGrm5ZkTfV3bU+WE4sJzOoH2LUL1znoxJcf3I5VmfWa3yPDttDje1dV4OntFTpEY77Ah3331/TilmXZgqIxlt77mwz7b6Cz7+dDa/EadLrint0qFhNuL770wHY8lsKxdUpUEjaqdmHlZAXhpm27Jz4FrP+NsEjI3iImDquq2q6qar7v/4cAlAM4DcD1AF7zvew1ADf4/v96AGtVVR1XVbUeQA2AC3SOm2zume2VeCe3Ge8VtMa9rIvm7cUFT+zRIar4BZ4XZbywi0QFTH96+eH3S01dn0jsOs00vdpRMOxralPVOYQv3rcNLf2swJ1IJjxenP/oTuwoTZCbSRalNekmluQcVVUTLqlnXS6nNKPEZcfD3ayHYdinjE8iPrSUiOdYkpse+6RjNPjsEqMhkk8fTymfmvmJ7Whi8H/Pf3m7AFcuTMeYQYnJdhTqELHDsbNqfz2uWpguOoyQytsH8aaG6dNJLnodG+yvGcfl8WJtThNkmfippX8MAJByuD3q9+4s7UTHgDnFhw63DOCL929D9/C4KeuzotxGe83kqPcDbTboOlAYN79yAD98LvIMvHqb8FX6XncourF1VVXjngHwL28XxPV+Il3peoEmSSdJatxGUSl/P/5leDmzA82mpeLwFEVRkgCcD+AggE+pqtoOTCYXAzjF97LTAEzvVbT4fkekmf/Gox0GT6ezwziVY9SFf60vMnWdq7MaTF0f2ZsNDsOQQrWZa3OaMeHxYnsJE0gTSceAE/2jLjyRUi46FArjH+8UGrbs8x7dhe89nWrY8mNlh/4QkV0ZcWM9lmu6eC8DQ70/VIJrfc9IbCsyqT1r6h3FM9sr5Et8sNsFexR+uGAfvvwgqylRfPQ8on+2NBtn3LNVxyXOFq4Jkq15iuaBhuFxN1odYwZGYz8T7skbTS4L3nASta/6V2vHM2daZTcqJK9ALVOXJWxbautRSzGMHHNZuLsKNRGKXkRDot1UkxUZ9UjedBj7AmYfsKI12Y346VJzZg9ctb8eE24vMqp7ADC5PcE/vnTM2h/5vYcme58q0LL0Opx571YMjgV/CJXIMqRrmKzWM7SRondERxCETvtnwZvAo58AHHywlmbSnDisKMpHAWwE8HdVVQfDvTTI72btyYqi3KEoSq6iKLnd3da/sCSKRc/wuC0GVoiswIjp3qS7jpBcTdcQipodosMgksp7hW2GLXtgzKVrEkS8bZ7o6beJaLYtxW348YvBp6Y3U233ML771B50DZlT5SmQ/6atVma3Zn94PReL02pRFyLBOdFvNotQ1z0Cp8t6CXN66h4eR0Y1xzNkkdPQp/syRTz4Eat4Krld/1ImLpq3V8doSIRwlSIzqrvRPSRnZUeewiet3l+P21cfEh2GocK1qfFcKutdyVI28SZTGzXmMuR0YeHuajy4eXKmxEQ8lvtHJ0SHELfpx15zn7kPEdn7yNVGVVXsLu8UHYZl6fmwCYdsw9Nl89j0PPGOr0LxoNMtOBKieBnxqCcb10kGbAcjO9+lm4xbdrSCnaB7qoGWvNiWV7LBt4yq2GMiW5qj5UWKohyDyaThN1VV9R8pnYqinKqqaruiKKcC6PL9vgXAZ6e9/XQAs67OVVVdBmAZAMydO9em3SWKVzxtfm33MPaWd836fXPfKE7/xIelSF55wDew1DDvGgBAUnIK/t83T8ezPz1X8zJW76/HKR8/Lq44DjX0oal3NK5lEIkQy1FsxJEvQXMSlF5h/WrlQeQ2xD9F2GUL5Jq68pV9tbj8K58SHUZEg05X3NM9ESWKRLxZR+S3r2oyKebqr52q+T1/fmtyOsCjBPdlVu9vQHuQhJ9Yj+lY+2ayJuC6PImdoEpyWrqvDkv31U2NZ1D0JL2MnNLab/0qvEPjkW9g13bHWHU+gRxuGcCPX8rEh485WnQoIV37Ysas3yUlp+DKr34KO0qZlCS7hz8oEx2CZdm1SrHs50i9trrb40XX0Dg+818f1mmJs+k5bu32qHh6ewX+75LP67dQmxB9v1H2YyYYd4KNd+v1aWW9F2Vnoo9vO9CyCQfHXBgYc+GEDx9jfEBEoehxvFfvBt68Kf7lWEFTdnzvj3osnu0xXporOgKyoYgVh5XJ3tBKAOWqqi6Y9qf3Adzm+//bAGye9vtbFEU5VlGUMwCcBSBHv5ApEejxpPxNS7LwxNaZ07Pn1Pfh4mdSsT6vJe7lG2VDlLE9/EEZ/u/N/LjW+dNXDuCf64viWgbN9NDmEry6vz7q94Xqj0qawyCcHv13PTat7N9PvNspo7oHYy6PPsFoZPRYzMCYC/O2VeDny+K8qDHB1x/eifMe3SU6DBKMT+1ro0LFrrJOJttTQnpxbw3+N87rAlnEelPmcMsAfrJ4P8bdTLQ10+AYz1FEepDt9kdXQIVWLU2zqB5YYOKc/xp9wnc+kP2aXXbpvurigeMCMm3XnuHglS9lSxpm4gkRTfd4SjkunLcXfSMTszoCRrUWJa0DuHHxfjhjGOvNbezHkrRazN9RaUBkiUXvU6hEp2QKwFM/WYnItiSjugfnPrJTYASU0Eo2Tv5br0MBLJkq1obTmgcMxJkztepKfWIhIqEiJg4DuAjArwD8QFGUQt9/PwIwD8DliqJUA7jc9zNUVS0FsA5AGYDtAP6kqqq52UZEAEbHZ+921V1DAICCJkfsy51wY9Gearh1rPrkdHkw6HTptjy7+Mc7hfjGY9ZMlnvtQCMeYbUMAJPHzIiGKj9WZLVBn66hcazIqBNSTU+6Snm+TTA2wS4KkVZWaPOcLi/+8HouVmc1iA6FaJaGnhEkJacgq7ZHdCghGdFDMKvtqOwYwo9fykR+kwMlrQOmrNOuFd60GhybvIbNrJF3nyaK1cCYC+Nuk68VLNDXCsUK/UQiQPy5m+O/JFPCPR0hauw6rXJyxsyU4jYs3Vdnyjof+aAUBU0OFLfEfs00IeGDmlbpi1gkTJKQqqpYklaLvpHJvkR6VQ+SklPQ0q/fbLI8RQVnpeM2o7ob+6q6o36flT4jkaG6yiO/xi6W/wB4/qvGrydsJ03jmWe4G9j/gvbX243eF5EJuhkptDmRXqCqaiZC9xd+GOI9TwB4Io64iKRyuGUAR/nS7PtHXViwqwqf/vhxuPlbn9Vl+dcsyrDt1IjbSzqQXdcb03vfLWjVORrxarqGMG9bBV7+xTdw7JzZUzzacfD4nId2QKbCj3qE0tI/ivvfK5E+ITqwL74sfXIA+pKzT8YXTvmYqbH8453CqF7PgQqympLWAZz2Xx/GJ47/kOhQpGfGua5j0Gn8SoiidLB+sk/8XkErLvz8SYKjmUlRFMt3RP033cPRY2YbwPxqgQ5fgm5PQPVP0aJ5AEvPm5lEZjj3kZ349hmfNHWdjb2j+M2rOVjyi29iaXotPnPCh3Ubd9KDzKcJvdp3Ck7Ew8d2xU0pRrC+W3PfKNxeFWecdLyAiGJX1OwQHYKpzDpmvvrQDnNWFMK63NnV3oz66LEs167nWRnaZL2uLe35DdF0eY39eHp7xdTP/pkgCpocOP0TH4lr2WYf4xIceuYzaRP/auXkJOAN864xZ4VEZAyrPBFllnfvAGr3Ap8804SVybTt9Y5Fps9GMtFScZhImLguHnRs9378UiZ+tnTmdPbjOlbPtGvSMADc+UYeq/5Nc++7Jdhd3hV11Wsr9w9lSRr2b0I9BgTn76hEWmU38uOoXm4G/4BP4O4jovjvluJ281eqs+a+UbxbEOe0LSaQYdA7EV37YiZuXLxfdBhSs/CpjBLYRfP24rEt9p9FwsiEIBE3mVMrg1dXidSnjrQVBsZcSEpOwfaSjtgCi1HfyOT06ysy64P+XdS5v7535nVsuO/6e0+nGh0Oke4O1vfF9X6nyxO2Il/goZvX2I+0ym5k1vRg4e5q3L2xOK71J4LOQSfeOdQ06/dWHkOh6JleHdwsvLY31MXPpOLSZ9NEhzGDlj7d/72Zb3wgEjDyQb0H3iuJ+7sPjC7RDtfm/tGpAhU0SVVVpBS3o6ZrWOPrDQ6IpBPvdz4h24yOCYTXFnrgRiSLaMkBnj5DdBSJozYVaD4U+XXjk7PKw+sv6MaO1BRVPbJ9iGLExGGSkvGd8OhPJsOSVxYle+NAUnCxJKIk0kX+uNuDu9YVokuy6nRWdePiLPzjnSLRYWhmt31dVVXkNfaLDiOshl5jqykOOV2o6hR3AWilcxGropFeWh1jWBkiWdMOzDhXaFlH4GtkPYfVdU/eBF61f3KfSPSmRtKviQCsyKjDjlJzE9ytxKwK2F96YDu+eP82OEYngr9AojbkUEPkJOk12Y0mRBKd3756CP/ZeBjdwzOveRO9fTaNBNu5b2QCZ9+/XfPruW8kJtn7LHat5iqrNdmNqO+xbyGX2fTfv/bXxDbDpGz0vu7801v5uGzBvggr1XedJD8rfeXsJwWny3naItuW+wARgLH4HiKnKPRWAysvi+INkpxVO8sA56DoKCalzweeOh0YiaZ/zsaeZmLiMEnH5fHq0jENdtrQdRAuhiCNboLZxAOeaeVts+usN4Ala5KErERtL6t8Tf2jLmzKb8X7RW2iQ4laTfcwkpJT0GDwQH407WbPMBOwRVp7qBk3LckyvcqjTH61MgdXPJ8uOgwhVFWFi5U1SCPuL7HjDQIie3g8pRx/XJMnOgxpNfWZkzjs96MXMkxdXyx++sqBiK/Jqo08xmL2w1u9I5PXaG7v5HmfldiMYdbXWtziQF5jdDdpvz8/uor2pW2DmPv4rqjeQ9qY2f8ubnEgV8MDDzLSeyyT/XeiSVYZrw+kV5EEuxUO8coyjaWOBp0ubMyTdDZDkw+gF/ZU4YrnIyTZk6li7p9YtfElMpSgA+OpzwLF68WsOyyLNRRaGsQl3wXW3GB4KJqUbJz8d6Qr8muZiEQhMHGYpNLmGMNZ923DO7nNokMhi5o+Degty7IFRhIeB3XloceNTVm+Tq3dPSv0CzcXTiY7bzMqSdQC24BmqvVN9ddscqKHTAqbHaJD0EUsbeaCXVU4675tGJvQNg2xkdOakvxWZzXgrPu2odtmN86MZIdjRlNl4wh/l72PPuwMPguOLF+fKk2vmEgubQNO0SHYXnPfGACgrnvywVMtrdHWw+14r6DVwKgoWte9tB83LYmcxD7dUIhzYzg9wyGqgFNczrpvG8razKm6dN1L+/H/NDzwICPZ+5uySeTtpV8XP4E3YpSm31uiyGq6hjHuDj1O19Q7iie3lsd0/+VdG/bRVmbW45/ri1DRoe+50opHeHPfGKo6h0WHQUQksRh6guODwI57Yl/lYFuUVWtFkOis12pCwQZNg/4SbROyLCYOk1QCp4hK5IEhsqdgp/fdZZ3oZSVT0+hS0VyW7AzS1Sv7akWHQGRvcTSdb+c0AQCe3l6h6fVmV7sjufhvMLU5xgRHog8zEqCNPGbi6TVZ4VD+xYpsrMioM2VdB0LMqGKF7UQkFcmPGV5tmuP/3szH398pnPp5XW4zrn1R/urQIoQ6ZHaWdeCS+alwx1hpdnFaDX7zak7sgVlIqHO13a5bilocokMIys7DeHb+bDILHBuO92sI9gCeLK3D3orOhJmFbVtJu+gQLMMxOoHLFuzDPZsOh3zNnW/kYVl6HSo7h6Je/pDTFU94Uht3MUFdDztLO9A/Ys7DYGaca2u6hqKeeYOIdOIy6R5CVNd+AnqCC74MzD/T/PXGhBdBs9hsbIHMxcRhsq1wHXk92s1YFsFTmDhZtT3YlC/fNEDD4278/vVc/ObVQ6JDsaSoLtgNPAB5bNvDvG0VKGkdEB2Gblr67ZEwF4jVDMWp6xlBUnIK9pR3CothdVaD5teOuz14cms5hsejr0JGJJNfrjho2LL9fZhYWtbV++t1jyPUz7IITBBQAeyv6cXjKeViAiJLK20bQFJyCrJqekSHEhe7Jb1ReC/uqUZj70jkF5pECXXGiGG3vHtDMUpazamWahf3vnsYDb2j6B+NLbnnme2VSKvs1jkq+zA7Ue/1Aw0ossksN4Fk7VsG4hl1NrsnRvuvL6I5/5x131b8bGkcFbej2NHcHi9uX52Lv7xdEPv6TCb7g6sh+y4W4x9rO1gXOtHR403sVm10wo2k5BS8kq7Pg8a3LjdubMjP6fLgiZQyjE7IP5Z6x5o8/P71XNFh6OayBelRz7wRKCk5Rado9OX1qgk9iyRZgMdGM9LIOEZnSIdews8pTAzbV8b9hIRi4jAlFJEDTZnVPXCM2fcpWa0aekYMTcwL9R3fuvwg7lpXZNh6o+VPfPN4Jv/133gLNXDERDn96NEOWH14z+rxx2N7SQd+sSI75N+vfTHTxGiM4W8tsmonqxLa5SaLXT6HlfmrnqYctkYFlnWHmrEsvQ4v7K4SHQoJYpfeU+CsLLJ482CTYcs26ruzyz5B9nDA11fbXd4lOJIEJEm/MtR1vsz93ud2VeG2VYlRIZYoXl1DTly+YF/YRIm2AWfIv/17Q7ERYc3g8ar46StZSKvswoObS3H9y/sNXyfNJnO7T9qYeZ3h8qg4WB9NVcrYdzArXj9FEzMPPTJSn68a7oQ79grDXq+Km5ceQGqFMdeMbo8XP1m8H9m+a9OD9X1YnlGPV9KsMTNjE5NRdWfEvehX0mtnzPRCpJuuCqBLwmIO7GAYyLdx3U6gcpv2t717J3BwWXTrCKatEOgzZ+bBGXRL9OXOScHNER0AUTjxdFBlenJ3yOnCL1ca/zSoFVzybBoAoGHeNWIDESTcQLDT5YFjzEZPtZGpNuW34BPHf0jTa610Q0LvgYo738ibufwYOtuqqs6qOEjk1z00jk985BjMOZrP5+kr+mPO5Xs4x/8vJQ4ZW+jtJR3IDlOJRxaGPKwWxzkzmnj0uP5ja6G/0rYBXLPI+g9mEZkpmmbT6fLA6fLEtb6RKGdncHJ65YRjx2I0ZlzXby5oQ3XXMF7LakD3UOgEYZEGx1w41NDPZA6Sxur99Xj4gzJc+PkTAcTe/oxNeOJK1ovWkHOysujTN30NP/vW53RZJmd1iM5Umx7FZgt8KYu3GCOWXXl1VgMu/dIp+O8Tj9c/IMGi2RyjLg9y6vtQGqYYUzxtRcegE/lNDuQ3OWb83pXg1aID6dFnlH2LGplTcSiqh12IorD426IjMEbBm0DuqhjfLGFr4zWgiv2O+4DxQeD3e4DT50Z+fdHbk//Fa9n341+GFlU7gLduBq6aF8WbJPzuyTKYOExSMXK4tqR1YKpSnh6iuRZzM2GFNLhpSRZK2zgtZjSEJYhImJEkU0VtPb2Z3YT/u+QLui/XP9YzMhHfTXYZ2f2+guyf71tP7MZvLkzCw9d9VXQoUXF5vGGnF9QTb35NenJrOb5w8kdx87c+KzoUMsH0B1d4CERmVFcr3uUGvj/R27Nw984W7anGR4+dg7E4ExpJfqpqrQcTAdhqLP2WZdkobHbEtYxzH9kZ1evDJdWYvWm17ns2+sqFiJRIxUSr8A419qOoxbjZ1ygyGR7+liAES1iyb7LK5UCcMzdesXAfmvvG9AhJkzbH5LpWZTboljhMcuOZb1K4a+JI7V64Pzf0juJnS7ORfe8PYwuMEkKCD8kQkdVoGUDzeoHN/2dOPGbRtUKvr+Ef9+XTOB06LlsiBWsm/20riPzaaC40HY2+/+EJlGZiKTSyrcA28toXM7Fg1+RU1byYINmoAJOGYxHDoLsRx//aQ01ISk7BuJuJEXprdZg3yE9yk+FGn1a7yjpFhxC153ZWaZ8dIcZ2VKbZIIw2PO7GsvRaeMNU6FiWXoe7Nxo//XAiS/SkTi2s1LZSaONuL8Ykexhqwa4qPLqlTHQYZIJE668nJadgWbo+U/eGOk3tq+zWvIzApGGXxxt1BWF3lBXFgsUt29mEiayz/e8bebhmUYauy/T377sGx7EmuzHCqxNXOcf7hHvo/VIMOuNLRJWV3S959lZ2ISk5JeqCMEYnDZtx3jP6Wk3L9XI8EYxOGFBdTgvZOiUB/F/r7nLrjR1OZ8ZYwpBN2+1YaGnq/7W+CJfMT9VnfTY/t9BMA2OumK+fjGgJwo2n60mP2XvILiTqPOSuFB2BNl0VwOENJq5Qou9ICJ3axZ4qfZZDtsPEYSIiEfznd63VcXihHjc9u5SByW9bD3cAAAZGrTKYlegdbKCpd1R0CKaq6BiMu1qLHVR0DOLuDUXwcLq1WWq7h0WHYCtPpJTjya0Vlr8RZFl2S4aV9ONo3cxaXhZ405GtdHQuW7APX35wu+gwKEZWT3C8eekB0SFELau2N673P7m1Qpc4Qn3zrx2IPQHzj2vy8NWHdsT8fi2Cxe1PYHZY5po48Wwr6Yj5gfVQY1L+9uvON/LwwHslaO5LrOvsSKzUJU2EccdNeS0hEyVdHi/+8U4h6ntGTI6KQvF/VW8dbAIAlLUnXgK+YU2ISW3TM9srzVmRRThdHvz5rXy0DyTWQ3fh2P3UEyq5OpqH3P1LGNXwoPCGvBY0JNg9D7NYqEsXtYaeEZz7yE68Hsc1aCTRPmjw1LZygyKZ6UsPbMfcx3ebsi5KNHG0Go44j0WzLuwWfxvY+Dtz1hWOER9Xxgv5xv2iIyCbY+IwSW1VZj2++dgu0WEExQpmcpLxXD7drGqL3I1MM5WrLfk+AgDXvZSJx1ihzVDXvpgpOgRDFDT145V9syugXbUwAz+zYFJHMPE0m3euycO63BY0GXxDW3Qfoc0xhg+K2qJ6j5Eh13QNIyk5BTVxJCdH23Yvz6jH6qyGmNcXL381q3G3V1gMxG5WLLS0BW2OMSQlp6CqU64HDszo45nZj3R7vHB5rN+GWKHvLUqom2f7a3pQ2madKe17RyZEhxC1l1JrRIdgmL0VXVG9vqQ1+n0t2LmiuGX2ctbnNqNfwv3D41UxIbiPdtOSLDy4uURoDFo5XZ6IfSr/9xxt9WoSL5HO0w9/UBayMnZBkwPvFrTi7g1FJkc1k6oC7xW0omvIOfW7xt4R7CjtEBiVuaw+U5GMD4TNOs5NCnHIKabicDTrnbVpDNw2u8o6saW4Hftr4nuAzYpCbddI/TH5jiY5rcioT+DkR3P2ErfX+mMzodT3Tj405b+OlCH1YXlGPf693pw+2XCUs/WQ5DwuwCPzdyrBAWZ51r5W0C7gczo1jN3J0ICTZTFxmOQS0AbWdo/EfBNKptOGWYOwMn1mUaw6uGjNqK1Jj21t9DFd3DKAlZn1xq6EbCnc9LAVHUMmRqI/tpPa/fSVA/jL2wWiw5iypXgyibkoYCptoxmdIK4FL9XF0KO9GBhzobF3ZsWxCbcXFR32rnSVUd2DV/cH74Ooqop52yqwLL3O8Dis0uYbeYz/cME+nHXftqje47ZBorHsBp2z24ZYLdgZvArbL1YcxDWLjH/Ibd2hZmw73G74eig0Gdo6Ix+o/PeGYvx1rbF90lDbMNz9kj+8nosv3h9d+6q3vMZ+Q6t66aWlfxRfemA7Fu2pDvs6K/Z5Y72n5hidYGVlC6vrnn0Oj+UBingEe9DYP87YPzqBv79TiN+sOjT1t0ufTcMf1+SZFZ40EiWpPUE+JtlYrH0Ap8tj+CwZieJw6wB6hseFrPuVfbU4WGf/ZHg9HpwXXWhEFI9XjfrhVgBYn9diQDRke4+dBCz5bvTvs1XHMzHbmsQWw/6boOckCm2O6ACISCwZq78kAp6O9RFLVz5w2ydi38hW10ACPPx+KXpGJvDyrd8QHQrFyO4DdbFNe6h9m8S79ey99SexmbW+H72QgVbHzGPpsS2TVcoy/3MpTv/ERwRFpq/AffUPr+cCAH570RmzXusYdQWtam8l0U6PKFJjDFOMPr29Avdd8xUDoiG/61/ar8s05qqqYkTDdLNGuntjsdD1k5Vp7811D4lJYggnlhvXiUqP9s5uLnk2DY5RFxrmXSM6FNLJtS9m4vYgfV+9aemHenxVu6dXHA4s5B24GLO7t5vyW/DCnmqk/esSQ/vWgcM2so3jWOiyIi7xfMcFTf2o6BjElz79cR0jMlZp2yB+wuFeTeI9BNbnNusSh8ysUMAmFK3V0+dtqzA4EjKN7yvXe5/Lqe/Td4FEkfRUiY6AiMhyWHGYbCvcoIYeU0bJNVQ1KZaY7liTq3scIsk+aBcqPhn3JyuIZvBSj12joWcEbo835LJGJjxoc8SSsGcuyQ8T6b12oBEpxdapzhZL4pHM4rlXZFbCmNFteiLewI/nm1ud1YCarvgrQ4STWtmFpOQU21ejtZp42ovApGFgsjogMJlAayarn7djaXpFJfhGSkiQfXaTQpMruyciI87B/t0uvaobG1lVxxa+9vAOXL5gX8TXuQOzwSxCstwtEsi/LyhTP5u3c+Q29IWdcccoZvcDzSJbUqbZ6nqMvV7UasEu+RMd7lpXZOg4k+xj+34JfshoUtczgqsWZui6zG8/uRvfe3qvrsuczsgZAK2yb5uhtnsYD2wuFR0GAMDl8aLB5HFWNh/6YVscP6Obpkc+kONYJ7IOdhiiY9MTQUwdR5tuCzIFE4eJElxLv/xJjlZyoLYXnYPOyC/003je56leP7F2uVv6R3HJs2l4envop6h/+soBXDjPuMFLokha7dym63i9bPU21YipVKMZaD1Q24uuKM51gYl2IoY+frb0gKHL31naCeBIYqmflhvwaZXmV73LrO4RNo2hGXhDTi4FTY6Y38ubQNExe3sFtu9eFbjh5f3YUdox9bsDtfaftlQvv16Vg3+uLxIdBulgyOlGtcEPLYkkU9Mc6kGTWRU5ARS3OFDXbd/vxQhaH5jxfw8/XLAPxS0OAyM64v+9cgAPvFdiyrrs4t2CFvxiRfaM38n+UJTZjOhLlbYN4NJn0zA4FjrpPLC/HCqOO17PnTWWz/6yXPT8Pgy7rrXwYd85OK7r/Sw9D5/AZXm9Kj4oaoPXog+K6Wl6IacDtb1o7NWWqGtG+/ZESjkueTYtunt6RDYW7XEX6cF/HlsknS13AXsfN3YdIavImXBi48UBacEbaBSAicMkFQ5WEgDkN/VjU741Ky39fHk2rlmk75P0ADDktGdFk1CGnC4MhBlQF6F3eAIAcLC+L2R/ys5JWNPtrejE3opO0WEkLKfLM2tfW3OgAVWdQzho06mfXtlXq8uNgUTpZRhdqbNj0InrXtpv6Dr0NuH2mrq+aL6D37x6CIMmn+d/ufKg4cnUdqTXobU+rwVJySmm75fxMnM8ySrtNcdiZ5pwe1HY7MAf1+RN/S6jukdgRGQKHgcJb/6OCoyMu0WHMUOw9vm6l/bjB89FrgRNR0Q69/v/7n9YTlWB+TsqDY7KOkTNoBDKP94pwv4aaz/Q872n9+Ini611Lfr8rmrU94zgYN3ktg922qzoGNK0rJ1ls8fi3jzYhBd2VxteNbp3eBxOl8fQdejhrncKkZScIjoMzcJ9a3p9pXK1RMaJ5XOOTszep5OSU/BvDQ/zRVrfmwcb8Ze3C7D2UHMMkVlHuP002Hnw58uzcftqeWZB9T/o2j86ITiS2dbmNKGsjTObhWO1S9E2x5i8szwkysmCKHclkD7/yM8vzgWWfl/fdch6nNuBGdfYIxKNZeuxLz18gjHLJVuZIzoAIquKpj1lQnR0frI4K+b3yrCle4YjDzJM7T++f4ec4W+yPbO9Enf+z+dx1FEyfELjnffoLnhs8DT+uNuDY+ccLTqMWeK9eeUf3GuYd40e4cSsomMQx39oDj77yY8IjcNsv38tF5k1PTO2/wObS2HX5qHNMYZ520JX+paR1j6CTA/JRNvidsRRLWBgzIWc+j5ccMYnNb9HsnvuIcV6ve3xmH/Oq+02dypGMeTuS4xNePChOZGf5R03MMHYrGOrunMIH5pzFP77xOPNWWEIrQ4bV+aXgCr5MUfGmPBY6yEIil/gze6XU2sFRRJaXlN/5BdR3CbHO63Z+g9PuPHx444xZ2UWuJaR7Tt851ATSloH8dgN58z4fUv/WNwPFWfX9eIrn/l4XMuIRTRjcdF+H8/vrsLFXzwJ3/jcJ6J8p3bffHw3vnbakZvOqmrMtUSoRWrdJpsKWvUKxb5kO+AltD6vBfN/em5cy+gamiz64C/+YLd7hGaMJZg5FmhE/ky88SdvOqxPIGGEPDexndDdyIQHF87bi/9c9SX87yWfFx3ObNO+831V3fjOmZ+U8t4mke56q4P/vqscmHMs8MkzzY0n7pMfG/C4NWdHfo3h/PtBFN9nsM5MI4sHkTasOEy2Fe60yocojrDXcAXweEq56BAM81ZOk+gQTKM1aTiW/TdwydHe4lJV7QN9lRorhlBsrlqYgYufSRUdhukya4I/7WiDXPugvAEnbWvelg7urnWxTUceaQvE1DYa2DkKHOt4PKUcNy89gHG3edWKjN5r4h3PqesZwbpce1ehMZPR/VteS0Tv8ufT8f35aVG/T7ZNLftDDLJtLy3cTDolspxo2hpRFVd5rtZHpG/Pytdmf19baNiyVVXF8vQ6OPzVC+PYTNtL2vUJKhTJ+jZ5jX3YVdaJ/2w8jDXZjbovf2DMhVuWZeP/3sif8Xs7tBleEwaFDrcOSN8fNosZ28Ft0HfqX6psVdFjpaoqVmbWSzUT4Kv769E1KE88VmVG2yziMLDDOYdil1UrUSVLzD4XjLu9uG1VDp7aaq1CLnpISk7B4ZYB0WGQLBZ/B1h0vugoKFDgSdT2J1UNHZVwnZlu++ZNkb5YcZiIbGV1VoPoEDSJZUCifYAV0gJFsx3jHQyd/vbcxr64lpVouoZir0xKROYQcXlt52v6aM84Ny2ZnG3h5rmf1T8Y0o3eN5SslmxjZoUkm9zDtr3A7ymWfSR502E8G2c1L5Hquofx4Q8djVNP+LDoUIgoDGudcSWisVm34mnbyAe9C5sdeGLrkRt0gQ/DRmN1VgOuOudUPcKyhJuWGFsRacI3k0dFR/Cp39lWULy07kN7KzqRVdNryfZTNlWdw3hsSxk+8iF5KmM+8kGZ6BCkYoXxPyvESLMZWQjDKLkN/fjPhmLMu+lrUj/AUd+TCDPVzba9tB1fO/2EyC8k60t9Ejj5bGOWHfLYtl6bJQ1R7WVrHpCzHLh+MXCUiJqsRu0z3BdpJlYcJqmY1ebrsR67NKcyX5jY2dT1bBSbf1M+p1rTQ7x7fHn7oObp3a97aT+KWxxxrlF/Zh/1Fz611+Q1GoPNpRhGnKeMHlPUkgzYPzJhbBA24vWqGHdboyKlXfqHVlXRMYik5BTU+Qa3jTrWZU/4jWVw3w6nOD3OF3UWvzFixZtm71l8OukfPLcP37VJX5dIKws2NdLKrutFng0eTOYuMZPLwy0iu8B2zMzxnvaByA/Xx9KnM3uvM2p9Ifv0kh9W0e5Dt6/OxYrMemOC0cisa4eytuCJ+npx+WYwGZ0wb1arWNh1XDnc+IiMH3nc7eGsNzpIr+pmZdYYjbk8eCe32bCq8nrRKzrmIJC09j0NbLg98utcTsArdx9jBrmbFut562dA0dtAyYbJJGKzxNR2qsCEte9tkDhMHCap6DpWIVFf9I2D0U2t9sB7JWiw+E1rCq6mazjm97YPODE87tYxGvlsL2k3ZCpCPfgrqEV7Qb8hr8WIcGbYGOU6DtT1YnFajUHRzCb7IIgW7xe18Qa5HUjUN7hrXaFhy472mnJswoO0ym5jgkHoza71mLrvvRI4Rl2GxKCXeD8j6WNzYRsAxL2/hKJ3xV2j9o/dZZ1Rv0fWXXXY6dbUfy5rG5yqIBePgTFj9h0KTdZ9j4hCs+JDCrK6ZVm24RVWY7Eqsx57yjsj9n3MnI1AVh0DTnRESASN54gx7XCT9LBuc4yhc1DHWaxC5aRK+vllZKej3jE6IfwejBlj/YFJW2Z/h5uL9H9QsKl3FH18IF8oLX2AsvbJpPFdZZ0oaZUjyfTs+7fjqhcyTFmXlftJkU6Lv16Vgx+/lGlKLLJwjNqzzbHuXkpkkic+Bbz5UxNXyKMyrFk3P026kNv0B2D5D8xZV9R826RqB/DkZ4DmQ2LDIUti4jAlJLMHA+fvqIzq9WuyG/Hnt/OjXo/Hq2JtThOfmJVY19B4XO+PZ3pDK7jzjXw88F6J5tc7XR6sy21O+BuX/1xfFNXr79l0GM9sj65dsrrRCTfW5zZreq3Xq87ap/76doERYZFJUorb0TscX/sbC1VV4Q2RON8r0Q2O9Xnajg1R3s5pinsZZp0lBnyDyLIWUwjWvlH0EnYTCtivf7nyIC5bsC/sazoHnfjRogw8vb3CpKhiV9Y2iEMNxlWWNHvXzKrtnfFzU9+oyREQUazi6RNY8TSY19gvOgRLeXRLGX73Wm7E18k+C4MZvvPUHnznqT1hXyPppQEA/a9b9P6sF87bi28/GX77RiXELmvGeKus14iyMqN1uWxBOi55Ni2m94ZKCIx0bl24uwo3Lcma+jneewWJ6n/mp+LCeZNtgyzHVqQwJAlTN9H0AV7YU41rX8xEXXd0RXWM6mcEezjZiHWFXib7T1Z02YJ0XZfnP12Eu39gB1Ybh7Zywj8ZqFan65Fwx0NbAfDYycBw9AVBZuAurI+a3aIjiF5t6uS/rbmA15vAN7EoFkwcJtsy+rwoY2e3qW8UyZsOY9V+sdNcEZllR2kn7t5QjIzqHs3vCTxyJTyUyQCPflCGf28o1vTaM+/dit+8yifyZBHv+bxvZAJ/eisft8+48W3sge9vV/7ydgHOvHeroesiscbdHjz8filKfFNvPruzSnBE4Z1571b8v1fkq2pnFL2PdL1vSLILcsTsbat9Yw85xVUJjvY7/NGiDPw0gY5BLTwm3SB7YXc1KjuGTFkXkRWcee9W3PDy/tjezBNYwojU9/Ff9/De5EyB202WpDYzzN9RKbyCqxaBzZh/XFGW8f5YojA7dFm2VTx6BDxgvnB3dcI9zNKsw8OF3UPjswoCOF3WKZ5T1Wnf65BokuxkrBIdWJE70YRqy0cn3HhwcwlGbD4DajSMOmfc8PJ+Ke8f2OE8TyRODOeWg0sBzwRQuze+VfPQnS3juejf88ZNvv+xUj/B9+W7ncCjnwBSnwSn3CGtmDhMUon3Gk1VVWwubLVFxd142ut+g6ZnJv3E+gQzz+PBaRnAiLdrF2v7ZKUuZbQKmvpRG2WlAFGireCxr6rboEgoXtG2g/4+QZtjzPTjcUtxu8lrnCTbk/Gh2k87VCjbXNCG1VkNKGp2HPldYSv8eXAyfkY736SUa8+3Fm47bRL8nqJhjL5+d7o8eD6guhsRAUUtsU0ZHU3vxorNZrlvWm3S/v3J1+ONzMxkiEQay+sdmcDvXpP4QWzfTi0ieS3a/eDW5dl482CjMcHEQUuSXXGLA9VxJkrOnoRY7gPJSse52aGO65Dg+60nduObjwev/CbbGFiglOJ2XPF8OraXdogOxRDRHJsyHyZGHMOy75vhvJHdhNcPNGJlprWLVHUOOpEZRdEhEWK9JtOb7OdZIkpw8Zyo9zyqXxym09CXCLw+G/ddh+Wu1D8csi0mDpOtvF/Uhr+tLcTS9DrDnxRN9CdRiWRi9xuXsrpxcRZ++Fz46cNlwSekrUuv0+30XcDo3UFLzDK1RTLFYpThcTeSklN0r5ThCbIz/W1tIT4oatN1PfH61/oiJCWniA7DdINjLjhG9UkKWLCzEj2+h1DseEbR8pkS4fInUgKr7N0J2eML5a9rC0xZj0vSB4xdHi/aHGOiwyDSTKZrq1YDjp2rX8jQfZlW1R2hsloi9A3s6KJ5e3HVwpnTbet5VJs1m4BVxHqcZNX24r53S6Z+rumS68H95v4xvJxaE/SccN1L+3H58/pO6W6kbzy2a8bP8Zzn7HTP6Mx7UnD7aokfBLCISl8SfXWnXMdwvKycFDudiE8hUVc6Iq/kwUaK7scvZuKXKw+aEku0ZEnUNfq0ZbXzosXCJZptXIeHEYY6gYn4Z4wwRsBBKvl5yhRT20CZ+bNbvpkeSF5MHCZb6fdVC9iY3xL2dXqcQoy+UWJW55SdYCJKBLx0SGDTznNmDVQF6yKoqopfr8rB3opOHZafWHu0nl+b2ZWlZPmqNuSF7xvb1e9ey8V5j+6K/EINFu2tQduAU5dl+clwLNvlZp9edpTG30br7aev2L9K7tbD5lTfEn3E5Tf1o3Nwdjvy5sEmXDgvzqkIiUw0MuERHQKZ5M9vhX+wQ4KujJRk7121OsZQ0TGZyCZ7rIkq2LGV19hnfiBh/PbVHMzfURn2GqmuexiVHfFVHjaDnuMEMlzjBYr1OPeq+szIJt8WsS+ny/g+mtPlQWpFV0zvdbnlfJCTxIm3On0k0c5CSURkLIv0ip77IvDq1cH/NtoH1At82NrsxCZhiVQxrHcqVt9+NhHuHGuRfZFMw8RhsqW67hEMjLlEhxEXCceYQkpKTol5sCBRxfr9MtE7OJmPFyMTFSfcXgyMWrutM4vM+whFrzumQT/V1Bs4gTF6VSC9qhu/ey03xvjl87WHdsT1fi1fR15jH49f0sysfpKMN4Onk6VqSTQCE5hl7fPOisuEfeFQQ7/m11rvmzde38iENJUPf7I4C5cvsMZsHUREIg05XYYnP5l5ZpDjLDTbwKgLEwmWxCVpF9NS/Ntw1PcQiTdMP+sHz+3DlQv1qTws2wOPgdcFsl6/APK2QXqRedtPZ8b3MGrAw115jf3Ydrh96ufHU8rw29WHUNziiHpZz+6s1DEyfUzff3qGx6Uf79Gb6Gqwlz+fbkrCO0UWy67f6hiTbkaEeA059Z2tkMiy2guD//6tm4HXrjU1lBmibayq4ruHaXl7HxcdAVkEE4dJKha5xreVln59pnV851BzVK+v6BhE15ATxS0O3aaQTgQJNm5hCDttwt+9dgjnPrpTdBi2FawiHJkv8ObQrrJOfOuJ3cio1lb1xP/+nuEJ047/rqFxfOuJ3UH/pqrAt57YjbruEZOi0SDGQeKhcWMH0hp7R3DTkgOGroNIJDv1ScTiVSQFNzrhxjce24VHPiid/IUEB92g7yaUluu6x7eU4YOiNoMjIjKWBIcdGch/GaH3WNXXHt6JHzybpu9CLcrIY+jcR3fiFysmp+9OlEQp0QlS8TAjcXbC7cWB2l5NrzV6UwbukVZ8ONJotd32Stai+Lk8+j8MctOSLPzvm/kYdLpQ0NSPhp7Jqcv9hZs6B7UXR2jq0+depBGa+0Yx9/HdWLKvVrdlhmonrXQqMqN7MBhHETArd1+Mjj2vUdtD6KH6F1riu21VTti/f/epPfBa7EtandUgOgSyg5gaeh1PDgM6zT75zq+A1ryZv+ss1WfZ0QqspjslQhvz1s3xrXdYvtkJjwixz0xvd0d7zAmFLI+Jw2QbI+NueKzV/0xoVy3MwPfmpeK6l/bjZ0uzRYcjjJUGCazOjts6o5odPiN956k9okOgIAqbHQCA4paBqN/rT9adt60iZGKvWYxOuo2G24AbC1r8bW0hrnspM+Tfn9paofs6zR6rtNjYKCUw7qtkRyPjk5WLtpV0CI4kNisy6/GXtwtEh0EUVHPfqOgQDJVVw2ttLYIlGug1TtE2IPeDvH9fW4Afvxj8WmZWJVIT4olVRYexU4TLxswE6YqOQWw3sQ+ix2ebt60CP1+ejcMxjLfoJdT47ds5zUhKTjEkMTLQnvIuodtAq0c/KIvr/b0jE3gju1GnaEIz+rgTNeYv4yX0t5/cg4Im7TPWROMPr+XixsVZGHdPXmPF9iBGdFtNj11nyOkKuw/6P0arYzKpOa1SW6GKeHD8ZSZuDmPctCQrrpmZJzxeLEuvnTrvOl0eLE+vg8erYsg5uVxHhNlQ2weccSWGE9mOZwJwTXuIpjYVaDqo/3pUnfrL5e8DG27XZ1kUHU39LN8Z1OtvZ0MlVxNFNkd0AER68HpVfDWK6bL1ueB0o7JjCGd/+mMhX5PX2I+dpebfpBRx4RnLU/8TvguOys7EGqSejoME5ot1LDHWQUg7JixHo6RVjoH2WA41Hp9yiPcYCvb+PRVd8S00ljgkXtmwwVNwhbqZsLtcvqd11x1qBhTg5rmf1WV5xS0OnHHy8fjG5z6hy/JEeWprOa746qfwzf/+pOhQwpJt+lpLiGKTcevOxq6CnFRVRX6TA5/75Edm/p7fGBGWpdci6cTj417OjYv3a3qdVc8dt644iLonf4SjjjLnE2zKb4HT5cWt3/6cKevTSyK3q+8V2qsivOyVeNsHxuC12O521cIMAMBlX/4UALnHB50uD2q6hlHdNTlG3xfj7ICVHUM49b+O0zO0KbvKJscPhp1ufOL4DxmyDr/fv54LAGiYd03U77XCbupPovzL2wXweFV858wT8YVTPio4KuuT5bvfmN+C8w0Yg/IXcPBYqDFu7hvFxc+k4pHrvorbLkya8beBURe6hpyzx/+t8/Fsw0r3YFZm1uN33ztDdBiaTbi1Jw8Gfg059X3Iqe/DnKOOwu3fOwOL9lRjcVotthS3oahlABv/90L0DGuvOE5EAJZcCDiagId9983X3DD578Ny3EfXRraLmhDx9DcAc4y5LpFOT5Xvf3wtuZYTq5VOvmQKVhwmWxDRtL2wpxpXLkwP+5qblmRhaXqdSRGRlagANhe2opcXVrpIxJtVZlTY0MOty+WoKG7FaTfdHi+8FhqMJeuS9QamiLju3liMuzcUa359pBBfO9CInyzOwqGGvvgCE2xpeh1uWnJAdBjSeCKlHD99JcuUda3PbTZlPaSN9MnpFuzvGOGtnCbctCRrKsFENm6P13LTdZJ9PLm1AnesyYv8wgh6hmNLKrOSRhOrKt+1rgj3vnvYtPWRudzxXNfzdAEA+O5Te3HRvL1xL0dkgnTcazYw9HvfPYxrX8xE5+Bkpe/uoXEkJacEneY8cBNO79JcuTAdv1xhQNU0CxCxZ2ntT4aKzZ8A6vZaY4w5mHMf2YmNeTpNx02axHNaivYSaDyKpMdgGnonZ54Ldl144+L9uPz5dJS2DQIAXj9gfPVts1jlvpFfPPf2xlwevHWwybR7P49tia/S+3RL99Xh0mfTdFueUUYnJguODPkKjxT5HiL405v5wmIisixHk+gIoifd+GVAz3bf08DiC2e/7IVzgefOjm9Vh1bG935dxFQWTfcoyP4iJg4rirJKUZQuRVFKpv3uk4qi7FIUpdr37yem/e0eRVFqFEWpVBTlSqMCJ3uSvboBGau8fRALdlaKDsMUhU0O/G1toS437EibePu2MiWK5NT34az7tsW9nP4R4272er0qKjoGDVu+WZ7fVYWcejEJf1+4bxv+n0mJaVakdUBQhiO3a8jJJ/BjIHu38OXUGhQ2OzS9tmPaVMuVHUPSVGh5ZnsFqhN45odY5Tb241CDMdN/TlfaNoB/R5HIbmXhEhCIolXbNXmDuLl/ZtKfLPvVF+7bhr+tLRQdBtncgdperMysFx0G+cjU/zOCmf32kXE3Gn2JQPEw9pyg3wap6R7WbVnhWPFh61iUt4cfp5JlK8TyfcQTe5HvutY/G9GB2l4AwOqshpDv8Y+TBiZ8+auSxkr2cYBQROw7GdU9Mb0v8B6c0Ye/kYsfGHNheYbc/R2z92mjvk89PoeoNjZYYmpdz8y+RJPvgTE9C9TEuslquoYx7vbEvN6z7tsGpyv295stnn12wu3Fve8exp5y82cXjNfzu6tQ3xN/n1YvofbXUN+P1mNFlr4Vkbks2qHVQrbOels+0FVqzLJT7or8mqE4i1YMtAKjwXIQYtnOkn03ZClaKg6vBnBVwO+SAexRVfUsAHt8P0NRlK8AuAXAV33vWawoytG6RUsUQrTNYHP/KLJqe5BV24NmEyuIRMP/pKtRhsfd+OvbBYauI1o/WZyFRXtrRIdhiud3T04bUNNlzg0Au5P5/oYRCcdZtbENDAd6enuFLssJZml6Ha5amIFB300H0WLdR17YU42bl4qrspnf5BC2bllEO2glowue2IMLntxj2vqibXUS6cEtPT/q/B2VWHtIWzVY/+5a0TGIKxem49EPDBrMiNLitFr8fHliVoQKp80xZsp6IrVjTldsVWL0bB+veykTa3MmqyHI2lbIEtauss6ws4ksSas1PIaiOBM1yDwph9tNW5fT5cF7Ba0JkxRGk36+PFvXiliyaXOMIaO629B16HXM1HQN4cqF6XjOAg/K/+XtAszbZtw4gR5uXZ6N789PEx1GBPq1t30jE5ytTEcix2Hj3StEdHnZd0gM/JrJrmQqBDNduEOub2QCly3Yh3s3lYR5VWRjE/okDluleRiZkOMelGyiSYSP9EpZxt7M0DHgxL4qY681iWbxH2RuZ/jXaaFn587RCDx9BjDUod8yjbT7YWD9b8xbnyfOAm3PfwV45swwL4ih8dX0/VvlDE9miZg4rKpqOoDANPfrAbzm+//XANww7fdrVVUdV1W1HkANgAv0CZVIPzn1fbh1+UHcuvwgLn4mVXQ4Qtyz6TDeL2rTbXnx9kEaekYsPS2WVamqioW7q6RNoNdTqAvb/tEJQyvvxiMpOQXPGJjca+TA8OFWh3ELj4LHq6Kxd0TXagFkHWfck4JbmRQ5ZV1AkmtL/yiSklPwgY79AT0ZMcAv+oZYu6/y8GsSTYEYanpTl8eLpOQUPL+ryuSIxJP1mNDqqW0VmqvTRDrKilsGkLzJ/KnTZa3SHtifdIy6pv7/D6/n4vbVhwBMHj9Pbi2HY3Syj9nYO2LoA1s06YXd1UhKTgk5NWsi9gaTklPw7Sf34O/vFMZclY6Ca3WMzWprM6q78W6BXNNlD4+70TWow40nyVy5MB2/WpkjOgxNOgcnz2laZqroH5mYOneI8EFRG17ZF/2DLmb2sRPxIZkhEx7IHtEpwSiR/XLFQSQlp8z6vVmJNnquJpqYRV9jixLqc3u8Kp7aVo4+SceaSZxIFc8TgRUeRjDjAZORMP3zkfHJc/7B+l7D45CF/HtFYgh1fPp/G9g1sMDhHLNrX8zEbausca1JNuSU8Hp3rA+o2u77QfKnCDKfB0rfNXGFejSG4ZYRzfJt3DCT4bRUHA7mU6qqtgOA799TfL8/DcD0jIQW3+9mURTlDkVRchVFye3u5lM7FJ4VLiitprVfrkTRS55Ng8vD7zmiGDeR/0nlwKlvmvpGsXB3NX7/Wq4OwYkTzWYJHPz+7lN7cf5ju3SNZ7p4p9BcEWS6Nb2S6Y6KtRdgIS/sqcb356dhf03iDLaF88Pn0gy5efGrlQexpbgNB+t60TWkc2JCiN1d0zOTKlDZORT2NU6XB9e+mIFDDcGmg7GmUDf57t5YPOPnivbJbVPREX4bGRVP5DfqGkZY2XW9uO6lTMOWX2XwNg4lp74PP34xts815pvWcBWnORcmniSD0fH4kj9quoZRZvAMKHoReanYFPDwXaPv562H27EsvQ5PpJQDAH7z6iHTYrJjgqBWKzPrABxpv2jSwNhkgrsZiWehPLujEk9uLRe2fiNcNG8v/vFO4Yzf/WplDv7xTpGYgEK4+oV0U2e+MIvI/TkeGdXdGJj20Emg8x/bhfMeNW58AgDueD0X63K1zZohWmZ1j9BE6tjpeyETqauTUd2NGxfv13WdFL3MmskHdF7dX4+7As4PQHR7RSyzdxnRJQ62TI/X2M63rBVCI/FH3dg7iqX76vD3dwpx+YJ9qI4wJiUDo4st8NaeuYza3NY8MiepKrC/pkdT8Zj+EP00VVWxvaQj5EOqQYXYaL959VDE/nkiHTdG3P/flN+C379m3jiMrLScUyONPXq8KraXxD5jkhX3ZVmLGZCN7HwA2PPozN/JfrDIHl/OcmDjH0RHoZ9EKvNOUtA7ZSjYHhy0FVFVdZmqqnNVVZ178skn6xwGWRXbQLKKvpEJOC1yQ7p3eBybC1tR0zWMZel1M/7mH+t1uj3oGnIaPvgrA7P7tnsquuJ6v7GDt8Y1uiIH+jsHnfD69uXsOvsmDH/jsV34w+u5UbVFtd0j2FGq/5QyGdU9+PNbBfjZsmxc96LONywNPmZruoZR0jqIhzaXGrsiHf1tbaHoEExhZity76bDKJ5WvezNg4263uB7KbVGt2VF4753D+Nwa3xPqdu/ZyCOzNXwL1uwDz9alDHjdyIv1ULdSOobmcCEW75ZS/x9arfvXzOvG656ISPyixLNtN3n4mf24ufLssXFIpDINuel1JpZ16Ki6fFwxJ7y+K71zNDcNwYA2FfVjdRK8+O18jif3g8hDIy58KuVObhjjdgHt3eWdeLuDcWRXyiBX648iN+uNibpQ+Z+WKBICTV3rStCQZPDnGBsLpYxwxUZdTMqDT/yQRk2FbTqGJX5ppruINvDP3Tc4XtYzewjySpHbnpVN6q7hoWNBUwXeC4eHg+fmD7odE1VPSXtNhdae+YiO+gYcCIpOQWbC1uxYGflVJLvuNuLX6w4iNtejb2CaFplN+58Iw+L9lTrFS75GHG/7q51RdhtgWs1o+kxK+jyjDrc+UY+Ug7PTB7uZnItJRrXmH7LyloEZDyn3/LMJHqQJ9T6t/4LOLzO3FgA45JOhjsn/y19z5jlW/qRMDJCrInDnYqinAoAvn/9va8WAJ+d9rrTAfBqiWxNxmrIfSPmd9jl2wrG+sZju8RMVRLDefybj+/GIx+UhV1cz9A4LnhiD+bvqIw9NoHei2JAPtK+mlXTI31SuF79cr2W8+DmEmwulOOmSEPPCL795B68kh799KpW0zcygV1lnVNJSVoZfdrqMKnSoYSnX7KR+94twZUL03VdZveQtQZUjRg6+PNb+dhXpW22GafLg1+uOIjKKKo1ix630ls8H8du2yKUbzy2C78zsYpMxEosHhW/W30I+U395gQURCJPi6yE+IKmdxma+8ZwwMYPlpF2GdWJNfvZbaty8FsTq5/r7eH3S7Exr8XUdV6zSN/ZKPzJK2ZMgW0WM2YOq+k0ZntZ6XrSQqEmpMVp9hl/mpqOXIJrCRnvucTCCh8jMMavP7wTFzyxW0wwJD3//hJLM2H04VDWPvng/N/WFmLR3hqsyW4EcOTB3qo4CgT0+q6zWx06Jo4FYXb7G2l1/r9/UNSGe989bHQ4Mcuu68Uf1+ROFZOhSbev1v7AYqgtNzoxea+0Z3jmWJMVzm9EuhpziI5ArM4SYM2NgFvwPS7pGh+D4un05fV4Q89YFZKmbSTbdiTRYk0cfh/Abb7/vw3A5mm/v0VRlGMVRTkDwFkABGTWkd1Idw6YxunySpdkuC7X3JspiepgvX2mtR/xXfxtyGtGUnIKdpd1Co4oOns1VPXVUgG3unMIt644iIffn1l5dHdZJ4pbHLGGJy29xqFeP9CIv60txMCY68jAvomDXFsPt09VDPMP3mVW95gdhjB2uZkSkklfop22omxTe8YSj9vjxZbi2KdBCyVkZbEgIeo91nz58/v0XaAGWm449I1MYN2h0NNV69nGbCluD/vg1c7SDhQ1OwAA+U39yKzpwUPvl+i2fukY2PBorbiv9aaUqqpYkVEfR0TxCbcbtg/o+8BKPNW1hsbd2FPRhTeym3SMSJzhcXd0U7FKJnC/sX2fKYIE//hkcauzGvDP9UUYd3swOsEqiJRYIrXfcl39AQ29o6JDsKVIffFoZuFwebwzqs6GfvgqcufB7fFiSVotxt363B+RIWnZStwG9dX99wrIeozq84dqJ2QS+NkNmZ3IpGuqeK9dJ3RqG/xR/OXtArx1UN5xjrdzmrGjtBPDvE6ImsxHtl59CyJLkP08e2gFULsX8IhKHPZtH08MibRG0rvjte0/wDOfj3N/4AAwRS9i4rCiKG8DOADgbEVRWhRF+R2AeQAuVxSlGsDlvp+hqmopgHUAygBsB/AnVVV5Vidb+9rDO/ClB7ajqnNoqrpTY+8I/vp2gdC47lpXOPUk7XSqqiKnvk/3ZBjJuzP2off3FvDF+Z8afe1Ag74rkki4fXVgbLLDWR1QAej3r+cieZOYJ6qD9Tn1Ot6O0vlC5NxHduKlveZPwfd/b+ZPTaeeiAkRCfiRAeg3tWy8h0Fz36jh1SaCeWpbedzLkGUsYvX+eqzLnZm4ujZMIqtVOUYlG9SY5u6NxXB5vKjtHp6qjOy/KWRmG3PHmjxc//L+Gb8zu113ebzIazSnWuz8HZWGTRt/5xv5ui4vvboHz++u0nWZsvrqQzt0W1bnoBN/ejN/qkKLrKo6h9AfpFLxOQ/twO9f016hRrRQ5zX/TddELz7kGHOhomNQdBhEUanpGkbPtGlwf/RCBr7yoH7ttBkCmyaZmqLa7mH8fW1B3A+JjBlYUEGm7aWV3tdZhxr6WEGPUBmhUuavo5gZ7/ev5eKcMH3eaB7+fSO7EU9vr8Dy9DrN79HCCkmK0/nHlM3217Vi70Fpkeitl7X25Mhk/D67fNfdgQnvgWPHWseXCpsds4pF+b/HaD5/LIUU9Gr7rntJn5kzarqGcNe6Ql2WRWIY9YCJGUQWMLCbYPdfKFoyngFDsVKsURq3+bjmwVeA0R7RUVACmhPpBaqq/jzEn34Y4vVPAHginqCIrMQ/RfwVz6fjlI8di5z7LsM9mw4jq1bs9Keb8lvxx//5PM7+9Mdm/H5HaYfuiQSArbsgCWlk3I03shvxi29/znIDtbGacHunkgkSpRqZEV/tjrIO/OWHZ0kzKJogu29U9Eq4NYvR1XP9h3t5+yBcHi+OOTq6CTkufibVgKgiW7qvDvdc/WUh69bbwx9MTrtz94ZivPLLb+Kqcz6NMYOS7ELtT+H2sqbeUZS2DeDqr50KAPjZ0gNIOvH4mNZv9g3/aI+fHz63D3OOUlD5+NXCB5ZjusESx/paHWPIb+xHSdsAdpo088L6vBasz2tBw7xrZv3t/aI2HHOUPCcxo45JrWQ5c0W7X4q+JtTqiufT8amPH4uPH3cMLv/Kp3D3VV+a+tu+qm6BkWnz3M5KpBwOXaXejIppX3lwO+675sv4xbf/2/B1xeqB9yYruAdrcyg+Xq8Kl9e6N2RldtmCfTjumCP989ruEYHRxEe2WUEA4K51RShqduC2C5Nw/uc+ITocU3UNjWN43I2PHhvx9ohw92w6jJFxN35/8ZlB/85xD/3U9xjYxkj0RYXq301NIqbM/Dkcfz8rXH/L7fFiTpRjLaHGZruHxpFW2YWffOP0Gb+/793DqOwIn1xtpFCX+kZfx2w93KHbssrbB9E1pH8VuQmLVozsHhrHyR87VnQYtmLULZdnfdeDoarsTrVpGpd3w8v7ccN5n8HCW86ftYxoxDMGH++m6hwcR6tjDKf914fjWo+ebUzIGGQZ8LEpj6pGTgaCnN9DPDOCGUFVVbyd04xrzz0VHz/uGNHhRMV//+XmuZ8VHIkGRe8AZ34f+NinY19GXx2w6Hzgzkzg01/TJy4ZDxKSgET7RaOvIE/24siv5f5MAaK7WicSpGd4HO0Doav5ydK0GTGworfmPvOrItrZC7urkd/Uj4aeEQw65a0keMTMoyXUjav8Jgfuf6/EMskOevji/dvwxzXyVVQL1r7pda/BiFsWovuaEt2HMY0Z27ypdxQD06qlvrKvFgds2D68ut8eT7GHOw5EPBgRzXG5UMKKplcs3If/ffPIQ1cH6/vwToxP6G8qaNUrLN35dw23V8Utyw7gvEd3zfi9nqo6h2ZVbwkZl/6rD+qmxVn4y9sFqGgXd7PZz+3x4q9vF8zY74wW6Th9dmelsevX+Lq+kQk093H6a720OsbQ66vk2Tk4juquYSxOqxUcVfRe3FuDuiDJhOFu1LYPjE1VWNfD6IQHD79fGvRvbx5sxC6THkggc/n3sH9vKMbZ928XGousStvirwbjdNkrKbsvSIV30fTub+nZvk6n9zVovYUS0a2cNG8lA2MuOEYNOkZFD5hFIaYEuTAf7390fOD69tWHcNe6oqk+rN+bB5uQa9LMMVYT6cGVkXE3arqGcfULGbgtisrVWmXX9em+TDP84XX57hNoE1tb8+r+eqRpmBEpnqF3sytzTz0MEcN7D7cOhFimsW25nrc20g14CPhnSw/gTyaOVUXLQqdaeSTg/bRYFTQ7cO+7h3HPRjGz0yaE0T7g3TuAN/9ffMupSJn8t2ht/DFNiaKBEX2jur1I7Ppj0VkKuLVch0nWaOl14il6ByheP+0Xkn1Osj35H6knAnDBk3sAGFcZZ2TcjeMtUGEiWqL7JYng+d1VU9M2n3XKRwVHE71I+4js0yrHIlwXrt+XGCn7+IJeVaCNqCZ9pDqJ2AbI5fFivcWm3rlqYTp+8e3P4VffTYrqfWYkgv7P/FR86uPH4uC9lwEA5m2rAAC89ftv49QIlQuMosfHzqrtmXHjfsgp11PsRnh1fwNu/94ZosMIq6xtEE9sLTdk2aGSyMK1WXomq5hdKaEtzIN34Rxq0OeG6+6yTpxz2gn49AnHzfj9Fc+n49qvn4qXbv1GyPdOfSUmnZQ7Bp0A5KiuKqIfEqlNrekaNieQECbPdQoumrcXYy4PK6bq5KJ5e4NeD6RXdeOMk2KrrC5S4EcJlyTx3af2AjCn+u5975YYvg4Sa2N+i2HL7hpy4pSPHRf5hSQ9Gcfo/CEt3VeLnmH9kiU7BpyGVGn8+fJsXZdnxGw87xa0GHLMVnSETsKXsZq1lV22YB9y779ct+VZ4dsxelypbcAZ9XtCXaN3DU0uy23ybD6xssKsdnesyTM9odMKugaj329lEOsu94ivGmWk6yOZ9+iInz2O4M3ux8nadBys1/9BAD37Y9WdQyErwCcqK/cTZfsq/UUweoblLyJnWV7fvZMhnaqdqyrgaAJe/g5wRxpw8hf1Wa4hdDxW1/0a+H+rgHNu0m+ZRltyITD3d8C1C0RHEp2CN4BL7wWOjrMK+bt36BMPUYzslylJttc/MoGXUmuQfPWXop5OPJQJtxfH6zSe/fiWMmmqh5h5OSDrhazekpJT8P0vnhz0b9WCEyq08HhVPL6lDHde8nmc9NHIO70VBje1kvEmnRZ2+g6MlpScgqQTPwJgMulNr8Q3s1R0DOGBzaVRJw6bpXNw9oDIrSsOGr5eI4/dW5cbH79sytuP3HCWtV1cJaDys1mbwu3Rv01/62ATzjjpeHzq48fiv088HkcfdeTT6JEMH88A/u9fz8Vp//Vh7E/+way/HWoIf7Mhpu9E1p1aYla5gfDMjkosS68THYYtBetq/tqASmN6qu4cwvaSDvzlh2eFfZ0RCWHhuDwqkpJTsOZ3F+Dis4JfM8ogvarbErMlGa2xbxQuj1e3cSWjZFb3zJqOneTWMeDEd57ag/t+9GU8sbUcl335FNEhRbSjVN6q6FYZE2lzjKGuewT/eKcIZ56s/8M3XUGux8kY0SbRW6M3HZzT5ZmR/BJ4OaXl8Muum6wE/so+fWetcHvCP8BrkabBEpg0bC8pxe2Yd9PXY35/UnIKbv325/DkjTOndrdyW+cXzfWhislZaj5+3DEzik5F0/RoHW/ZVdaJMZcH1537mYQc1tKzPf9/rxzQb2E2YeV9SrZzvVXGUKXXVwec8NkjiZYPnwCc8X3gtvd1XMm0iiSl7wKuEaDgdeCKx3Vch4523AcM6FwIq6da3+WZoUXu8eigMhcAc44DLvmPvsu1cuNNliT36DglHC1N4GNbyrAysx7bS3R62khnKzLrUdEhfopjYPJp/khJGRQ9GSrRaZFV2zPryfi9Fd1YkVmPBzdrq3plxHXZlc+nI6u2x4Ala6OlnZHtgtTP5fHiRy9k6DbN1FFGVBz2/SuqS9vQa9705f4njEXLb4ouQVrW/TtaNvkYppt+U1DGfUFVgaNsfE2cWaP/+e/edw/j58uz8YPn9mHBrsqYl9PUZ8z0x62O4FWPte5/ZiT+FTU7DF+HCFZJtNFiZebsBwqMmHYzEo7ZyeGnSw/guV1VGJ2Y+XCEvzKdP9Eko0rMNcee8shT/Ir061U5+Nd6C05bqLO3DjbhsS1losOIyEZNecKo7Z58qHyJL4lut69NED0rT6AtxW226ivEQs+Pf+G8vfjlyskHU+u69e9Xe8MEK9mulTAOtwzgkvmpGIowq8zUNyfhF3XnG3n43tOpIf8+7vbgyufTcaC2N+RrWvpjm+UmFP9WGgsx5uZP2jH7AbFY+aNsHxjDwbrQ25HkY409bLZIbZIWbx1sCv1H3/lIwiYtomjP+999ai9uWpIFwNiEwT+8nou/vl0w43cdFq14HQurHmt2w+9hth2lHbj+5f2zrpm4reIw0gMsOh/Y+u+Zv6/fB1Sk6LeenffN/l28F39GXjsfeGlyG+jJrtf6MvY/RqMYfy56B1h9rYYXGv1Bbbp/UMyYOEyW4/LNMxJuwJQm3bYqBz/lE5ZxGXK6MDDqwoa8FhREmZxnhKY+7UmRty4/iB+/lDnjd/7jZsLtxXM7K4VUFKjsHMJDm0tn/K6sbRBvHmw0PRYrmN7SdQ+No6x9ULcpoQbGXBjRYTBxukS68fjU1nLRIQAAntxaITqEWTbltyCvUZ82M9TlUWZ1N5KSU1CjY7X3F/fWoDmKdtaK/O1+38gERifkSH6frrJzCMM6t0tahLrhESrpVVaH6mM/7i5bkB7ze4fH3Zi/owKuCFWhppt+tvjKg9tx++pDM/6uZ3LN8vQ6NPaGTuCIpn9lhPk7KmYkQOr1yVMr5U5enC7ar9vp8khfFVekSMeiY3RC9z6gmcZdk58vVLfTP0tCWXvoad2JACCtUv6HgldnNeja36Xo+NtTr3eyovjzu6o0v1f2a+M/v1WAopYBQ5btdOt3nTEy4UFScsqMmVP0cv3L+1HaZsw2IPubv7MSDb2jmHBruwZ64D1tRRzM5D8Phrr2qu8ZQWXnEB5+vzTo34XwhXrvpsNISk7BVQvTcdkCnRMtdOQYncCCXVW4ZH4afrYsW3Q4wszbJt/YZby0XsKKeHBIjz5I7/A4xqaPG/o+hr8q+2sHzLmXE13hDv37XoFFoiTv3lnSL5Znw+PlhjXKNx7dFfbvMlfRFf2Q0F/eKkBRswMTvmtCf3OeU9+HnaVyFreTntN37RUsSXbtrUCX757rSPdkJeKxOO8xZi8GitZO/n9nKXBoZezLyngWGGjV+GIJjqtxu46JSrBt4/HuHUBDhugoiGZh4jBZjhGnAys+HRsr4z6rPS/svv7ITpz76E78a30RblycJToc/GrVwahe3xkwlaH/699d3oUX99bg8ZTwFZaMGggJTPz/0aIM3PeuOQPoWqbllWlvVlXg58uyUdI6oHti08b8Fnx/fujKIhRerAmFDT0jqOocQmnbAFr6R3H76kNxVU2MNpEgnv27qNkxq5J5MHetK5qqyGCU/CYHACCvUd/K+hc/k4r3CrQOAFiP//v/xmO78JeAahaxyKrtwZBT34dQth6WZ+Dtonl7p/6/uW/UkIQF2WkZu39+VxVeTq3FxryWoH9/cHNJ2Io1oxMe7K04kuQ6vRJTNH2RUN3cJ7aW4xaJb9K+nFqLP67Jw8+WHsC4jok2fSP2mXI28MYnHyINL9KDIec9ugsXP2PdPmCoa9ojExGK3T9yG/vwixXZEafYtpOqTjlmXYpWvNdXE24vrg94WFdvh1sHcJ3B66DQ3i1oxYHa3qlKsi+n1mh+bzwt0ca8FpS1he53tjrGwv5dNCOKGHxQ1Kb7MgEgs1rcjFjRsFrXJ1yFWruQ/eGA+Ii/WRKqkrE/slRf0nNFx5DUD9jsr+nFoj3VGNeYYG4EGe59veKrwp+IRGz+ah2OiW8+vhvXvzy7D+ofDzdr5qbAh8yDmapEHqFZjqbV1mP2gFiOPZmTOI3SNuDErrJO0WHYlh5VyIWRpKvlb1umH513rMkTEkug7LpeDOp8b0aoiYC2dyD4vYaodPnyIOpSgZS7Yl9O3mpg/W3xx2OW7MWiIzCGDB3bQLFcF/bWAl32e7COrGuO6ACIYmXrsTmdvVvQgq+cegLO/vTHRIdiObLtZ/FWBgv8OJEHLY3ZALXdI0hKTsG/rzwbf7r0C4asIx5DYy4sSavFbRf+t+hQAAAH6npx7YvG3Cz2VwnQm4x9d1lc8mzarN/trzlys/J7T+/FeZ/9L/MCisL1L+/H8R86WnQYMxjRThc09eOG80/Tf8EmMfPwu3X5QXz/iyfjtdsvkCKeWGkZmDcjye73rx3Citu+FfP7jUiWm15BS1XVoNVy/BVg/FX5fvRCBk786Iem/v56kEo04Y7dny3LxjM3fT3qWMN9+lCVrH/wXFrMN4SGx92Yc5SC446Jv13M8CWtVLQP4ZzTToh7eYC2ZAaer6Njp83VN2JMH9BMd28sljIxqaR1MpmvfcCJz37yI4KjMccVz6ejYd41osPQRVGzA+0D2h8QNKpi63QyzhKRKFKK21Hc4kD/aPQ3Y71xVE775/oiAJg6ropbHGjtH8PVXzsVwJGH2+xy3GmxOC1xk86A8A9Nydg/+fnybBQ+eDn+6yMfivxii9I6FlFt0YdrADH5OpGqs/L6xZ7Oe3Qnrj7n06LDmGEsqkq3ctHrQduqTvFJ+VlRXO8Ffmp/e+H/faxVbZ0uj5DZ0eLV2Bv5IUlZmlS7VBwecrrwoTkWrdkn281wALvKOnHPj74sLgBZDpBp+kYm8ImPHANFUTDkdOGWZdn4zpmfxNo7vis6tJmcg8CcY4F5nwO++Vvggj8AL34DuO6lCG+Ubz+cwWWt2SltScK2KiYvfkN0BEQzMHGYpKJl4CfwNa/sq8XHjzvGmIAEK2p24Fwdksf+8c7MGw6J7HDLAD534kdwwoetuc+4POZ2iIzuf83fUSll4nBdzwie3l6Bp7fzaa9Y1HYPY3OhMdWAZGLU8dHSPxaysooMRkxOXIh008gml4mmMWK/tWqVQRntLu9C15ATp3zsuJjeb/R526sCRwc5JANvxpTpUJn57o3FcS8jGFVVcaCuF98980QoihJXFZlzHtqBM086Hnv/dYl+8em2pMkHjy4445P47xOPD70+izaiz+3UPlV8QrLo96qVvxlKKW4XGkckVj2+Et31L+8XHUJCK2jqxxc/9TEcf6wcQ9b7qrqjHj86Uv08+O9jcd1Lk/vl/uQf4LT/+nAcSyKrsuIpZUJghVUzaH1oc9BpvWQzmZNzE7EaZyT1PSM446TQ13xW4Bh14e2cZtFhzOCI4aGhYNw2SYiU+cgzesaZqxamo8GXhKvnmt451DTr4TQ92994ixBR9L728E58yWIFvLTuc4pi/hhHt4bZY80wMu5GRYf4+x8NPSO45Nk0PHDtV1DTNYyLzzoJAFApQWyzzPvskf8/uGSyWi8AFL8z+W9fHbDhd8BPls18n/QDaTKfDW1O6gsUDbHV7gUqUvRdZjykP9bIbBZ97InoyMXgvG0VuPfdw4KjMcb1L++fquCml0Q/D/z4pUz8yjfFpFU1xzGdqsTdKkNtLmxFq8SJmHZS0TGEf/uqI9ndnoouJCWnYIwVwELKqe9DTn2foevwn9ecLs/UlOBuj1f386ddJHg3wBK2FMmdCBdMbDdvVbyW1RCxaks0+2zYKHwLejunGbcuP4gPdEo4rOuJf/pKo2zKb8X356eJDsMQKzPrRYcgNaNvnIqiqipGJyLf/IymTfJ6Vby6v17TcqM14fHaPmmKSAut10uO0QncuDgLf1tbaGxAUZpetS+q1lWHptjjVWdc1zy7ozLie7xeVYpr1CYNVe5Iu3DjuZEethXFnr2RI7SOsZv57ZS1xfYAZ6iZSsy8j1DaNoCk5BTUS3x9JaumOO4VkD3ZLcE+t0Hb2HKoTx1vW9ZgUJ/mPxsPY962I4VrBsZc+PaTe3RbvqTdA9uTIcHULlxefcZTvj8/FU9tK4/6ff5D6E9v5eOGl/fPGsOOZ4YZLVr6R5GUnIK8xn4AQKPvfJ9W2YW3c5rwf2/mA5icWTgpOQWbC1sjLvOGl/cbc//WNQZ4PUDTQeDhIDPpuYPcny/ZAAwEPjQk+RWEloY1fw0w0mV8LIlm6kawQ2gYQWnpaKy5ETi0wvhYiGLExGGylKe2lk8NgOk5cCXzhfRjW8p0W9ZDm0tQ1yN+eiHRik2YStRINy6OvQJStIeN5F30GVr6Qw/g/G1tIYb4hLVp8pscokMwVcegM+5lmHqsmXjn5+alB3Dz0gNxLUPrGfpLD2zHz5dnAwBufy0XX3pge8zrtFLbF0y4G8dFzQ48o3M19eEIFZRkGqgOtfvLFOOjW8pC3riNxP8ut8eLwwL6O9l1vXhxT/WM34Xatj3DE3jo/VLTHzZp7Ju8ER2u3yBSrN+91bG6vL5C7Ua9w+OWTuR682ATvvLgjpCzH8TSlu8s68QjH5Th6W36zzRy2YJ9OPuBbbovl8hqvvzgdk3J+aO+Y7ukVbIxmyhOQlWdQ1NtlB7nrr+uLZhxXeNVVRQ1O8K+59EtZfjyg7FfC+nlmhczRIegiUzXAeGxNyQbGbvt/aMTcb3f/5mOVE4370O+mx852QUAxviQOFFEiwLGZfQgsnr6/3tF29hyYIvlb9P0bMmiGbOJtotx7iM7o3yHfbQPsOgPzeZ0xZc4vCy9FmmVXWjsHcXSfXVRv99/nVDou/4KnBE4lmTkaOyv6QEArM1pmozH9/vAZsh/Hb1gV+TZ2QqbHVif16JbjFOe+DSw6Q4g79Xwr4vUhlZL3g66nUBnmLydgVbg/T8D5R+YF5OdSHh9pcmgtuuY6FhmoIJsQo5534g0Wpp+pGPnVY1/mksG6/Na8MSNX9NlWa8daNRlOSRWz3D4QWCvV7XQjQ/9fO/pVDTMu0Z0GEQh1XYPW6by9cPvl6KxdwSv/vYC0aFENP1G1qGGyaev06u6RYVjCYvTanVdHh/M0N/6vBbcPPezkV8YwH8DY/6Oyhn9Zr14VRWKFzjqqJkdDX+/Y+vhDmw93DHzbwg/5rOtpCPMX4G8xn5Udw7hrE/FN9Xf8IQb5zy0A5/75EfiWo7RzL66sUqfUcbEDJmF2lzfeWrPrBsdslBVNWIC+Y7S8O1FLMZck+ewgTF9piIOlEj7rpbvkBLXkNONj3wo9DB0dl3vVHLxUcpkpd2jJNmdxjVWDnd5vLji+XTM8QUemFgSy+GREjBDwubCNmwubAv7nnW5ckz1PiQwwciOEmAIXCptjjF85r8+HPY1WpNqzTw1vlvQiou+cFLU7/Ofv/2fScTpnLt47MIlMnq9akL1R/Xm9aqzxj/8ZN6sKYfb8bLoIEzkLwzlNWln99+XDrVvAMC2w+347xOPNyWeWGVUd0tTVMvNjo5wrQ5r3LeKxpNb9X9AfLqN+a0YmfCgd3gcS381V/flBx6f/v5ZqD5osKM5v6l/6vp0unDnt5iVbADO/Xn416gBD4AFJtjmv65vTHrrqQKWfBe4vxuY86HZf/caM7aYMDotOsN85Vb9l2n4BRnPuzQTKw6TZLQ3gv9aX4Qz79WnIT4sWzWT6SzSbnMASh5n3rt1aoqSeB1uHcCjH5TBwwt3orj98Ll9+PWqHNFhaDqtrM5qQGqlNZJvef6ZTY4hX4pHbXdsM0SoAKo7hwzr25513zZc/cLs6nHh9jk9ksi0VkIItypVBYbH3Shr13/2Er0lWtXhRXuqUd4e27TKFB1Zk4YBYEdppw5Lib29kXfLhKeqKqo65ZgK9Y1sPqhMsbtlWTZuX50LYLLv8Pl7t+Led0sERzVpwqMtcdg/buJPfBA1jJJg3YiE0TcSXyVZis6eisjTC2s/xs27Qt8QYwW5WQ86wP8ARNwhRRGDeeuym4feL8WQM3iiypn3bsWtK7JNjsg+PigO/7AOiVffM6K5r6aHLcXtOPPerbjqhfSQr3G6PPjfN/PxC8HHXqSzz69W5uCXKw+aEgvJL1LBKgo+7vzWwSadxrOAzkFn0Pvx/p8iJfoHG4P/yeIsXPfS7FmM9cqviVpTQAX5nfeLiSNegQnQUyxyZ845CDjkeOCYiOTAxGEiALmNfaJDCMnMKcHi0TM8LjoEmiZS9T6/SF3YJWm1WLW/Hgdqe6d+l9fYjzoNCU1tjjFk+aZSSXQFTf2iQyCyLBGVbl4/0IiBUT4drBdZqlckgoImBy5/Ph2lbcYlYVZGmaDGbz86iXbDXFVVLNhVhYxq9hn1ZMXk8+Ew1etVVcX7RW0Rq376r0mjmeZUa2LMmgMNuHW5fEkXq7MacMXzoW9am+mVGKb+JArnbd+UrDIJ174G/ilwPI8FuY/YV9WNHzybJjoM25B13wp2uIxOuLH1cPvsP8hGS1/Ket2tkEIlpPg3gxn7mFXugciosXcUL6eGnllKlgrw6w5ZL0Fl0KBZSWTy0OYSPLRZjoe1ojXkdOHSZ9PwbkH4KcLbDKikWtUZ+v6Yv/JxP8eWNZO0K0MSGAzxYIwZnK7JMaip/pDB6/vX+iKs2l+PrNoebC480q7Nus4M0WVSMHmd1TXkNC7ISIreFrduGch6YRZo+aXAwnNER5EYVBU4vCGGN1pkXyLbYOIwEYDmPnmn4LBKodeiFomrNgchtONsUe8XtSGzugc3LcnCD57bF/H1ly/Yh1tXmPvE8oTGKUTNduPiLNEhhCXrdvOz4zRJs0ja1g+MuuAYjf9p86X7aqceONhV1omk5BQ09o7EvVxgMlFy7bSkgqbeUV2We7C+N/KLyHIeCHEzpKJDjmqNfs19oyETU1weL9ocYzhQ24v3QtwcGTD45tZTW8unphOPxKzxspb+0anpIrWQNbHSMTohzY1dms0q479y7t2xy6juwV/fLkBOvbYHfhti6AsUNPfjxT3VIduGBzaXIqtWvr5BsUTX4QnRZ6eYRXPalbmtDdfVYNLbbLkNwdvtB94rQV2PPteD8Spp5awHRgl2TNy76bBus6SJpvWYN7tN0/M6xx+7HtfL/RGqZkt6eRaSbG2+xyv32C4A3L2xWHQIUZPrWzbGawca8doBa84cMjYxs+JjYDviv37rHjK36JEs7ZmVxpZk7v+TWH9fWxj1e9bnNuNQiOuQeARW9NV7RhC3b5awZel1+NvaQrxfNFn13t/n8K8+ZOKwAty2Kge3LDXwoff9LwA91cYt3yoczYA3VNVhC+itER1B4ijbDGz8XfTv87AKPJmLicNEADbmt2B7iZzVDkIN9nUNOZEncaXkwmYHblw8e/oLWYy7wg+mHajtxc1LD4R9jRUFXn9HM4bx17cLopq6aGTC/E7zxvzYpuQzQ6bElfTWHpKvktN0F83bi15WNRfi3Ed34rxHd8W9nKe2VUy1qe/5npbWK9FlY34Lkjcdnvr5f+an6rJcScZ4YyPRYGteYx+6JTp+rVLVdOvhDqzMrA/6t/vePYwL5+3Fz5dn4+/vFJobmM/S9Dq8knakolGwqdCm/qbDDrksvS5sQlqrYwzfezoVi/Zaf9Drd6/l4vzH4m93tZKouSAKyciHIfIaJ2cGae4bw3O7qtCo0wNIZpH1IQiiQNEkWMmeOFDSGvw6RmslKKPJlMyWWtkV9Pcyfcf+G/JWJtP2jKS53xoPmbyUGvm6QusxbnbC2vKMGGYA8H0Wt1dbYkosIl3jsE8TH6sUn7Ea7pZyC/x6ZPm6ZIljU4RKzESA9nFBUe1hc1/0YzT/3lCMn76i3/19sz66/zrO33fs9xf0ibLisWEPaE6MArseBFZdZczyo9FbC3Qcjvw6o7z8LWDvY0H+INmF2du3io5AHzV7gInIM2FLaSzWXC5ZehOUKJg4TFIROdAZbErnuu5hvHVQzoS6axZl4qYl8ia2Pri5BAVNDtFhhBTpIueudYWaK1olgt3lnUF/7xidwMupNVFV+DOSzJVzo53a3Uwj4/I/GWl09UwjNfeNYvX+4AmAIoga5JlVBULDe7Jqe7CnPPjNZqNZ4ebAutxmVEnctgDATUsOYFm6vFOXN/eNIik5RXQYQYWqjLC3QswxEWjco+2c69ap8tHrWQ0h/9Y1GP1MEmsPNev23etdYcJMbkn6cKQvK5zDAm0pbgvZlzdynGBN9szqWh6LbTxrRUukzVGSZ0Fe+2Jm0N97A9qPwOaElW2PkO0bPhCkonx+U7+01wlW0TM0u48ceJzIqnNwHPO2VcAV5ppH1s+SU98f9Xv8CSY9vod+s+uMmWVBVdWQidRWuiwZGHVFLEpiNln3R6uzckJ7m2PM9rNuyvr1yLTf7AlxX48Si9Ml//2/UPQ8mgbGXLFtC18QHWHGnzOqu7E+tznGyEKsNsSHj/Sg6FGGXWz51uuS4IH7F78BvPI9sTHUpc3+nWxjGZU2uJ51NAFv/AToDD6TqDQccuaTEWnFxGGSisjrqWDr/vGLmbj3XYFPLCF0p9jsagXRkqxrNItMFVhiESxRd9DpQkFTdIPDWpO738oJ3uG5/70SzN9RicwaOSo4yjxIKtOAUSArHA8yRxipvfvlyoN4+IMyU2KRmcuj4g+v56K8ffKGeauGSkO3Lj+If64vMjo0y7p7QzGueD5ddBiWlh/ledNMIQcHZWkQp8URbkxMrxvA4RYTyzpadKx2JvODS5GMWzh2s71+oAGLU2sjv1ACB+uNSfYwUlplN/7ydvCpy/WoXK6VNG3sNL9YkW2LJBtKbFk12tsl2ceTQpld9U7MASpTO/Zyai1K22ZXaA43W4UIwaoOby2Wc2Y6K/nr2oJZv7PSeeuVfbVYG2I8FJB7nCpeVZ2TFb3CJcjEYml6Hb71xG40BZnhQeYx3UDnProTP19u4DTgMbDQ5rOUBovNRjLdhfP24oIn9uDuDUVYlm6N69j/396dh8ld1fke/5zA6DMz9zqj17k+8yje4AxeR2VURGBcxwER1AuuGEEvj4OjKHBFVAwCEkFCRFZFCKsJSzYIhED2hQTIvnWWztpJOulOd7o76fS+d5/7R1Ul1dVV1dWdrjrfX/f79Tw8VP+6uvubWk6d3/l9z/c7UKGvaazcm/6amKW347R1Q5vIGEJxRb356+HWWS7qMRRaO7ozFuBI9pHfLtTXHl454N+fGGu2Zemg+b0n1+qXL2zpdaystkX7B1H9N/VcyUtaUXLk+Eb3o03pC1gkxp6CbsQ9sFLqjEZHEURUR54qaA+1B86UytYO3e9rrhm635VOPV0J0BuJwzCjrbM76EX3dC3ImjvC78KL7KKPsYsAqTK1/46KdIup3//LukGd9OQi03uzub1L0tBVEzxZ3YavQFheADcc2nHex5KvG9vsVR7u7+FryKFackeOlTuHwtYMrXXzraO7R4u2V2lfTexE7/fzd+r59WWDajdVGNmf2ZLqJrMVqBrbugr+Ny2Pv9lYHv/WGO98MHdbZUErXmfrbhB6c062i1VltS16YUN5AaPJTUdXj4mLh5bfg6l+83JxZFp9Xjtlk/bVnGjhNndrpdnPrGQLitNXQzJ+apl3K0qOanKWqus4Oe1d4dd9RoKBbAa0llSaq9TPtFCfcdY+Wn/x/JY+x6LwDEfpZVjIDTYDsf9Is654ondyZeh5+0C1Zagq29jWGal5bMIbe2oGNCcc6vP8Zbti3XPKj/VdB4raw7nzsK3uT5bXnaMs03vg9Jvm6PGIJOLNWF+u8XN3hg7jpL2WpvtW6su+0NcNLn98jbamSSS09HZctD0aFYfHz92pirr0yYdf/uObuuC+5QWOaHhpas98rcD6uVdJdVO/97lx5hZ9a+KqjK+hZDsqG/SBW+fpj0v25BxDZ3fsTZ3LdbUfTF6vN/bEEv4+c/dr+vw9y3L+Owmpz8ja/bW64ok1enBxLOY9GR6TyrrYhq9cE4cnLh/kunBikKvdL/3lYumV6wf3e4Cc2B6jeln9iFS6InQUuWlkozR6I3EYZnzg1vnmdmpj8Kx/jD+96kD/dzIsXQvdLeV1/f7c0SFupZ1pDWTn4dxbcN4ya+iqeltp356OpQWjVJku3Gw7VK/bXrbR/mNHZYMeXrZXZ45bGDqUAbO2+GIpge2XL8QWdaJo4fbDoUNIa8OBMBV0x87smxAQBZYvrtWnbDpo6+zW9dM2Dfln+WCV1bYer3hdiGSFbu+14UBt2s8sy3nr35y4Ur8wWDn98Tf2DYuLh8isIWkTycPL+m6SjZLCzqRsDijziw+nTU6wloB1zMhn1EBMXBaNpI8bZhSlTbQajmydPeXmxY3lem5N73UmW+/OcLz32lJe1ztJ39yT3PfZsnYeL0mPDvbifkArUqqNWz7/SiddvIfqWnXmuIUqKqsrfEA5KK6oz9iGe/LKsOvhiUSWdK8Ca3OaqInaeyvqvJfW5lDdEkPn+5PW9TmW+rrPtbvmUDrSnKYSrrG3Y6bPBGteyrJRO3WNNKQVRrqvDsRwrzi8Pd7hpDlLgnSyts4e3bdo98D/TmX/170X76jS954cXNXRxJA2KiV7q7oxlhDcX/Xi1vh7PdciRRPmDXRdOOX8qD3+eFQVD/D3DGf2ziEjz+B5eUbFL0qTvpRy0Gj8jjRR9MYrAkBeROlzPIrSrQUWusLIK5srVNXQnvZvX/TAGzn/nmdXZ277N1AtBqqEZ3Kspe/iRllti+ZvC598mGlt+duPrtJkI0n2103dpHnbbO6Ae3Z15seoo6vH5GnBXXN3pD2+4UBtwSsSHk23wBoBFq/JrCg5om88kp/K8/153lBC+kBYfB4zmbu1UrOK+rZRtqAQ876/rCjVNx5Zpelp2iyGvkhaVpu5osWRDO3jQst1MX2kGg4VSJPfF1YrEVpk9XOhpLpJd6aZv1mLN4pJ6s0d0RgPX9x4SLfMsrGpc7A6c+2yEqEha3NZnUaPnaMbZmzW3fN39fpeoecnbZ3d+sCt84J2c0unrLZFlzy0Qr+ZFa0LyhZfhndluLgfpTVYa59b/elKs0Ox3GzXpJjK+jbd+ELfjb1n3bFIi3eErTyZeK2mvg66unsi28XIiqg8fFelSf607JmUNd8/LdmjLwyTyqcHjjZHphptJlY/U7J1pQrhVxEp9vCHBbv6v5MBVzyxJnQIQ2bG+rKcumWG1tndk7UTXWLD39ZD9Vq972jG+1mX+BemruElKh4PZM5/zp2L89u5qmiK1BavwOwLeP7ZeFja+kLh/t5ARenELDJ4TPNi1CmhI4AxJA4DyItcW2FgcNJegCrwQ37d1E3akcMOy0JqaOvUTS9uNZkMk67ty1f+9KaufnZDgGh621vTlHbnubX3sbV4Ep58c3/aKtvFFfV6/y3zzFQHTfZohl3mj7++v8CR2E1mWpihXbplh3Jox4XebC3lZxeVC4H5ti9NdYXQicOXPboqY0tMmyMc65j9+Y97luvVLTYT9XOVeFuUVDeZSyIbqEK+XqM21Fq7KP3UitLQIQxryR93rR3d+vVLQ9e9pxDOuHlen2PVDW2qaey9kdDqeV86k1eVZvxeoacnVQ1tauu0N943xzd4b07qkhWJZzgSQcYcOGo7kTVZ1M5p0iUxWaxGnWpTWe9ORN571Rpam0qdv/zzzfM0Y300NyNbEZWKzUsMdwvMxb2LdmdsER81n79nmf7r6fWhw4i+NG89a591VivkR1lFyhp8Y1unDhzNXg3WmuKKet34whbdGIHE8jNunqfrpm7K+P3EzOyGGZs15rHod7bONNUcSHGK6sZ23TY7Txs3u1qlWT+WXvxR7OtCJg4/83Vp5lWF+3vZVGR+TZrSFLG5V09KrkIEzr0iyZE4jN5IHAYwZN5/y4mLQHyM51d3j9eGA8e0u6pRkjRnS2XkEwKGwr6aZk1de1BPvFH45MfBsNJiaVZRhW6YUdT3G8beyMbC6eXbj/ZdkNhcVh8gkpNjLfkkpBc3HdLszdFKGrP8HrEqKhfXpPDJsVakfc4MPDSXpfkckOyubUUpKSuEQ3WtunZKRBaAM/JqaOvUBfct1674OUMUpB+XC/d6jdpQ22PsFJBqgfmV/OhOXXtQU9YMXfeeUM4Zv0SfuHNxr2MlEUrIsbQJMkqf7dYSL6eu7dvRwtJzO5xE6fwrIbXCnbGXb06sPOyJ95WVePqzJkJVC63NCbM54+a5em1XxJJYhqHhMG23MJakrqWXVDdqfWltoGjSYz1x6H1ywtJeX1/26Gp97g/LwgQzSIlCQpY2FmUzZ2vmbqRRnJulFX+rbimPXVfcbqlw1+Ft0qGUIlhN8cI7hUwcbjhUuL81KAZfjHN/eeJ2T49UNDVcLLl44F+l7uTicAYf0+GAisNIQeIwEAE/eS58RdJcdHT1qPxYdKpcvLGnRnUtvU+K6ls6deec7bm30Ayksr5N33hkpS68/3Xtq2nSNVM2hg7JlOQFG0sXJd7ccyR0CBmt3Nt3MdzcdNzwCkC6JHDD4UpKH3OQReMMj5OFXfr/b+ombc5QlcHS2JIQ+gL8U29GY9PG6LFzJMWqslmsUJ/O/G2HtWDb4dBhZFTIV17qW+8Tdy7W5QbaBGbqwmAh6eSx1/t2PQgfFfLNe0VmjEuW+h6vbe5QQ4aK3vnwxQde12UTVyXF43XfQrstUy1u+uoyfi4dZcnzT3vPfG6ONrX3f6cIsXTOZymWdJKTVoyHatq54xfrP+5ZJkkqq205fm5jWUl1k3736nY9s/qAdh6OzmamhE0pawJReP2W1fauhGglaezNkti6qI1o+vftCFUtnL6+9waI9q5us9dpOru9bn9le+gwcjZ67ByTa4CwcS6U+tK44L7X9cNnbF3PTd1YMGXNwUjMH6wbPXaOxjy2ShOX7zXXmbU/Lxcd0rJdNaHDGLAovW6Tu+qkVqhOVn6sRe1dfbvBmjPxU9Lkr/Q+5uNxHyngmpkjtW3Ats86cXvLNGnW1cFCyUlDubTqodBRDI2maqnd6OcDFYeR4tTQAQDo39ytdhNFUiXaIlq/YCFJ33tyrSSpdMKXjx+bMH+npq61X7XnwvtfP37bYitKpPfdJ9f0er1ZFzoJMVWmBE4rNpfV6SOn/f3xr0fZevj6SE3quPzx1WkTyEPYW9Ok8+9dHjoMSZmTrrhm0NeUtQf1n58+PXQYOTt3/JLQIeTs6mdtXXRIVciPi9S3Xmp7c0uueW6jOgwk0I2fu1M//Ow/Hf/6lllbNS1NZb1Cq6xv1cEItdaOmrbOnkhWkerxXqOSUnLOumNRwWNYm1Qham9Nk/64tKTgMeTK4nxkwryduuUrHwwdxoB57zXf8CYdqffz3R2l0oJJPv67xZE6J85m+e4aNRZwY0N/rJ2/p0r+TDIeqiS7MVY1tEuKzX9NVSHL4oL7bJzbD9Y3HlmphT/7rH7y3EbNvPqTZl8b2VibE5KEmX83zNisOVsyV2cMLWqvAe/tfi6MZBZeRhZi6M+hulZ1dPXoLafGEu5SNxpg8Fbvq9XqfbYqTOfip9OKQocwpCwUbUiV3FXnkxOWpj0H7uzu0ad//5o+/O63qbWjO+x8bc1jUvk66RuPBwwiB9Y/jK3H13osdAS5aUm6Tm39Mc3mnjNCR5AZSfhIwSsCSHK4vi10CMOGxYl6f17bVa2FxbYvFKZjYs5kIYY09tY0afLK0tBh9NHR1WNud2xL+4ldrT09XqPHzklbkRaZ/fqlrcdvL95epTdLbCThZpK6EBEqaTjd8FFl6PP48ifW6NI/rwgdRlZW3qsdXdFMYLFs0faq0CHkpJCJKpYvyqRuNMjWRq/QklssP7v6oLoMZA987g/LIlPFa8K8naFDGJTU1tpRcKTJVptMA7n/We2taQodQh+JaoJSLCnE+jrLvppYl4tXtlTqx8/Z7uSTXFWt/FjmqkVR0NXdo+pG26+NbGqbO3TlU2u1oNjOXM3ossxxUUoSu/ShN/XIsr4dGywZPXaORplYEBwZHlyyRyXVTfrI7Qt11eT1ocMZMAtVOaPqNy9vO76GG4V5TcJrO6tDh5BV6dEWrd5ne9002ft+Pdd0J8FczDW0PjFYu6sa1dPjVdXQple3VPQ67wjFwtpKLu5btFuS1NjWqdaO6HUmwsmraWxXbXOH6QIMuapqaDu+3lXT2K7uCJ1nJEt0hNh2qEF7a5q1/0iADpxtDVJ7ozTvl9LWGYX/+wNm7Pzn3n85cbutXlp2V7hYcmLs8ctk8zSpoih0FIO3//X+7xMaawlIQeIwkOS8u5Zow4Ho7Q60xDlp5+EGbauoDx3KgH3/L+t0tNnWhepcRGmRrdC++ucVGmew9VmL8cWZqJ5oh1ZccaLazw+eXq9XNlcEjKZ/Vi7ctnf16KpJ69QdX+xpbu/qVenPAuvVrj/y24WSwp/2J1exXry9StdP2xQwmuxuTkr0t+y/nrZ/UbqzwFl1li96f+i2BWbG1lRWLma1dZ7YqBSVzQaV9a2auNx28lA6Hd3dZtpSD8R5d9mqBm/tMWxIqW66tybAhaUBmLSy1Nxzmmrxjioda+7QkQhcQO02MpYPhTte3a5z7rT92simtdNeO1nr1328j52DXv3MBu2usrfpIXltbXN5NNY0jT/lw8rSHSeSMOtabGzc7c8ts06c8xqbzmjc7GLN3FAeOoycPL3qwPHbM9aXmZ/XJERhfBjz2Go1tnVq7X5ba4CZfPfJNaFDOCk/iW+Q6+nx+sHkdYGjGZxVe4/qgSV7dO74Jbp2yibdMmtb6JB094JobDJOrGmcOW6hyXkY8u8Tdy7WWXcs6lUJN4rKj7Xo3PFL9KelJWrv6tYn7lyskmr7r+l068UmCrBNOE266z3pv3d0r3RkT2HjSTXtCmn/Gye+tnbS25h0DXjRb6QNk4KFkhNrj18mzdXSY5+L3Y5KzMkm/5/QEfSvMfobyjC0SBwGUuyuapL3Xn9ZsT90KL1MX3fw+G0rlQXTOf/e5brogTfU0mHvIkome6oazVV/HYjfGkyMtaKxzWaCrsUWosmt1K1dUEB+WMo7WLKz+njFseunF+mBxYEXJPpxzXMbdW+8UoMVPT2+TzJRoVXUt+mZ1bGLaj94er1mFdlNnn9uzcH+74ScTFlzsKDJw9Y/o2YYbfloZdPSB26dr5/P2Bw6jJw9uHiP+Y1Amdw5Z4epz/qBWLS9SqPHztHRprCJnHfP36mLH3yj/zsW0PxtJzrkrArULaI/TUnV10N1tBioj92xSLe/av+8uq6lUw8vK9ETb+wz+/znan4Euz0ls1bRvafHm13/SNh3pFkT5u80+9zvqGzo/07G/CACmwyHC4ubBfrz7OoT57zWNkKVHm3Rz5+PzjlBwpqIJLhKNtee0zlz3EJd9uiq0GGMGBsPHtP7fj1Xi3fYrkidiXP2qmknupccbWqP9DVGK5LXF7u6e9QYeK0b9iQq/7++p0ad3bbmN9mkTsV2VzVqtvX1xj+dJT10tjTu78LFUHdAmn7Fia+d4dS2zgh0pbD8+KVT+qZUsyt0FMNT7b7QEcCYiI0OGG6mrzuoScYSdGdtOqQdlY3mkjF/NXOr6ls6deVTa49XFsTQ+ML9EWgZECFWK+1Z0dbZra1Gq9ckduZaruZoXWV9q9YZq5abSfLzvOtwY8BIYhJDx87DNi/Yri+tPb54Ocdge7/7F+82MXe531hC9XDwmbuXhg4hqz8t3VPQdq1FZXVqNbxB7VCdzUXCj96+KHQIx83cWK7yYy2hw8jJ/Yt3a/zcaFQQSrW3ptnE5/tgJDbx7qgMG//DBtvUd3T1aNPBY5Kk7zy+OnA06ZUfaw0dwrC183Cj7p6/S7+bs0N7IlBVKZOfTS9SVYP9Cs/ZWEvCe3hZiS6MwNrWo8vtXpyimxeGM2N7HY6rbe7Qj5/dEDqMnI0ynoxbVnviHMt4qAjk6w+vDB3CSXGyNwdL2BmBc+8oVGX97hMnKnv/dHqRzhzHtXD09s2Jsc0mh+vbdPBoNNYWJfW54nrh/a/rF5Y3UW18OnQEfdXslpprQkfRV2td7P+RSMqN2ARx0pelqWNCRzE4pStCR5CdOyV0BDAmCiMYhinvvX41c6vGvbLd1E7MNftrC95yOVe/mrlFy3cbnBRhxFuyo0qSVNfSoWnrwlfac3J68k1bmxISbnxhi9nWZhfct1yS/WqOlv3bXUv1rYnRqJTxb3ct1YLiw9pw4Ji++ED4i8ytnd3y3pu9EPPNiat0zwK7u1vnGklmtjqHirKyWtsJWO1dPaqoK1yMRWV1unHmFkmxzTjW2BzBYp5ZVar2LhuP2ad//1roEEaEqyOUiJEsUaX21S3Gq68EcMusbfrawyu1/0hz6FCy+mX8Alhtc0fgSGDRS5sOhQ7hpLV12pjzfvT2WDLFwu1VgSOJvgXFVfLem5xfAoNVVFYn773ZjZePLt+redtsViFPdde8HXphQ3noMLL6zN2v6bzxS7R631HzVeiBwbj15WIVV9gsOGF5LSghce3HsuTK7nO22Fjrhk2H6lr1pT/a6hCVjfdeOyobtGxXtZ5ZVRo6nMwObZRaaqXZ14WOpK8/f0LyNs7De+nplrq7pB7Dc6+NT0s9PewsK6RJXwodATAgp4YOACPXQ0tLQoeQkdXPTavt/IBnVx/U7756pn4+Y7OWGGgXNXdrpXZV2dzlbb39TFN7l256cWvoMFAgP3rGTkLR+fcu10/PP8P0QmtRWV3oEDKy0oaysa1L42YXhw4DBdTY1qXNBa6kv760VgePtuizf7CXfPrgkj362Rfer90G5yG3vlysW1/m/YnoCLkh0XrC6+fvWRY6hKye31Cu977jb7ThwLHQoQB5YWHjpSTVtcTaN1s5F4i6qyav11IDa1rAUCmrbdHKvUd093ybm6BHjYrO2GW5Ynqyww1tGvOYza4UwHC1fHeN/uqU6Ixn1o0eO0e/veRDocMAhlRFXZsufjACic6Pfz50BH31dNuudnVog7TwFumIzfm2pFgi+OzrpHedGToSWMEaElKQOIxgZm60vUMbwMAdabLRbtRq0nAUfPi2BaFDwAj24JI9et87/zZ0GBk1tnXp6w8bbzFjwKSVpaFDwDDX3N6lPdV2P+ub2rsi0S4cQGZ3z98ZOoTIu3fR7tAhACPGZsMbHKOEpGEMN9dN3RQ6hKxmF9kurgAAuXh1c4W+ftZ7QocxrNxGUQoMMxaLX0RGR5PNSsMJU74VOoLcVVE0DAkkDqO3vCUOO+cukvSgpFMkPeG9n5Cvv4VoKj3aEjqEjC55iKQgYKBGj50TOgQAw4Hh85XtlTbb4UlSSXVT6BCAgmlo69JVk9eHDiMjNuEA0Rey2jEADMS3Jq4MHQIADMqhutbQIQDASXt+Q7n5jjUAEGmWE4eBKKLiMFKMyscvdc6dIunPki6W9EFJ33HOfTAffwsAAADDx76a5tAhAAAAAEAkrCs9FjoEAACAEW0JXQsAIH/ueGfoCIDh5WPfDR0BjMlL4rCkcySVeO/3ee87JE2TdGme/hYAAAAAAAAAAAAAAAAAAABS/fXbQ0cAY/KVOPxuScl9HcvjxwAAAAAAAAAAAAAAAAAAAFAILl9pooiqfL0iXJpjvtcdnPuhc269c259TU1NnsKAZeO/dmboEAAAAAAAAAAAAAAAAAAAGL5OOy90BDDm1Dz93nJJpyV9/R5JFcl38N4/JukxSTr77LN7JRVjZLj83Pfq8nPfGzoMAAAAAAAAAAAAAAAAYASoDx0AAMCAfFUcXifpDOfc6c65t0gaI2l2nv4WAAAAAAAAAAAAAAAAAAAAgH7kpeKw977LOXetpAWSTpH0lPe+OB9/CwAAAAAAAAAAAAAAAAAAAED/8pI4LEne+7mS5ubr9wMAAAAAAAAAAAAAAAAAAADI3ajQAQAAAAAAAAAAAAAAAAAAAADIPxKHAQAAAAAAAAAAAAAAAAAAgBGAxGEAAAAAAAAAAAAAAAAAAABgBCBxGAAAAAAAAAAAAAAAAAAAABgBSBwGAAAAAAAAAAAAAAAAAAAARgDnvQ8dg5xzNZIOhI4DQbxT0pHQQQAAMmKcBgDbGKcBwD7GagCwjXEaAGxjnAYA2xinAcA2xumR7X957/8h3TdMJA5j5HLOrffenx06DgBAeozTAGAb4zQA2MdYDQC2MU4DgG2M0wBgG+M0ANjGOI1MRoUOAAAAAAAAAAAAAAAAAAAAAED+kTgMAAAAAAAAAAAAAAAAAAAAjAAkDiO0x0IHAADIinEaAGxjnAYA+xirAcA2xmkAsI1xGgBsY5wGANsYp5GW896HjgEAAAAAAAAAAAAAAAAAAABAnlFxGAAAAAAAAAAAAAAAAAAAABgBSBxGMM65i5xzu5xzJc65saHjAYDhyjl3mnPuNefcDudcsXPup/Hj45xzh5xzRfH/vpT0MzfFx+ddzrkvJh3/uHNua/x7f3TOufjxtzrnpsePr3HOjS74PxQAIsw5VxofX4ucc+vjx97hnFvknNsT///bk+7POA0ABeSc+99J8+Yi51yDc+565tQAEI5z7innXLVzblvSsYLMoZ1zV8b/xh7n3JUF+icDQKRkGKf/4Jzb6Zzb4px7yTn39/Hjo51zrUnz6olJP8M4DQB5kGGcLsg6B+M0APQvwzg9PWmMLnXOFcWPM5/GgJE4jCCcc6dI+rOkiyV9UNJ3nHMfDBsVAAxbXZJ+7r3/F0nnSbomacy933v/0fh/cyUp/r0xkj4k6SJJD8fHbUl6RNIPJZ0R/++i+PGrJB3z3v+zpPsl/b4A/y4AGG4+Hx+Pz45/PVbSEu/9GZKWxL9mnAaAALz3uxLzZkkfl9Qi6aX4t5lTA0AYk3RiDE3I+xzaOfcOSbdJOlfSOZJuS05QBgAcN0l9x+lFkj7svf9XSbsl3ZT0vb1J8+qrk44zTgNAfkxS33FayvM6B+M0AORsklLGae/9t5PWqWdKejHp28ynMSAkDiOUcySVeO/3ee87JE2TdGngmABgWPLeV3rvN8ZvN0raIendWX7kUknTvPft3vv9kkokneOc+0dJb/Per/Lee0lPS/pq0s9Mjt9+QdL5iZ1qAIBBSx5bJ6v3mMs4DQDhnK/YIuyBLPdhrAaAPPPevy6pNuVwIebQX5S0yHtf670/plgSXLqECwAY0dKN0977hd77rviXqyW9J9vvYJwGgPzJMJ/OhPk0ABRYtnE6Pp5eJmlqtt/BOI1sSBxGKO+WVJb0dbmyJ7EBAIZAvL3ExyStiR+6Nt4W7qmkXWKZxuh3x2+nHu/1M/GF33pJ/yMf/wYAGKa8pIXOuQ3OuR/Gj73Le18pxTaBSPqf8eOM0wAQ1hj1XpBlTg0AdhRiDs3aNgAMjf+UNC/p69Odc5ucc8udc5+JH2OcBoDCy/c6B+M0AJy8z0iq8t7vSTrGfBoDQuIwQklXMccXPAoAGEGcc/9NsXYV13vvGxRrSfFPkj4qqVLSvYm7pvlxn+V4tp8BAOTmU977syRdLOka59xns9yXcRoAAnHOvUXSJZKejx9iTg0A0TCU4zLjNQCcJOfczZK6JD0XP1Qp6b3e+49JukHSFOfc28Q4DQCFVoh1DsZpADh531Hv4hbMpzFgJA4jlHJJpyV9/R5JFYFiAYBhzzn3V4olDT/nvX9Rkrz3Vd77bu99j6THJZ0Tv3umMbpcvVvHJY/dx3/GOXeqpL9T7u2NAGDE895XxP9fLeklxcbkqngLoUQroer43RmnASCciyVt9N5XScypAcCgQsyhWdsGgJPgnLtS0lckXRFvlyzvfbv3/mj89gZJeyW9X4zTAFBQBVrnYJwGgJMQH1O/Lml64hjzaQwGicMIZZ2kM5xzp8er9YyRNDtwTAAwLDnnnKQnJe3w3t+XdPwfk+72NUnb4rdnSxrjnHurc+50SWdIWhtv8dnonDsv/jv/r6SXk37myvjtb0pamlj0BQBk55z7W+fcf0/clnShYmNy8th6pXqPuYzTABBGr0oOzKkBwJxCzKEXSLrQOff2eOvmC+PHAAD9cM5dJOlXki7x3rckHf8H59wp8dvvU2yc3sc4DQCFVaB1DsZpADg5F0ja6b0vTxxgPo3BODV0ABiZvPddzrlrFRtYTpH0lPe+OHBYADBcfUrS9yRtdc4VxY/9WtJ3nHMfVaytRKmkH0mS977YOTdD0nbF2sVd473vjv/cjyVNkvTXkubF/5NiicnPOOdKFNuFNiav/yIAGF7eJeml2Pm6TpU0xXs/3zm3TtIM59xVkg5K+pbEOA0AoTjn/kbSFxSfN8fdzZwaAMJwzk2V9O+S3umcK5d0m6QJyvMc2ntf65y7Q7HiGJJ0u/eeCvEAkCLDOH2TpLdKWhRfB1ntvb9a0mcl3e6c65LULenqpLGVcRoA8iDDOP3v+V7nYJwGgNykG6e9908qNp5OTbk782kMmKNwCQAAAAAAAAAAAAAAAAAAADD8jQodAAAAAAAAAAAAAAAAAAAAAID8I3EYAAAAAAAAAAAAAAAAAAAAGAFIHAYAAAAAAAAAAAAAAAAAAABGABKHAQAAAAAAAAAAAAAAAAAAgBGAxGEAAAAAAAAAAAAAAAAAAABgBCBxGAAAAAAAAAAAAAAAAAAAABgBSBwGAAAAAAAAAAAAAAAAAAAARgAShwEAAAAAAAAAAAAAAAAAAIAR4P8DpjANG7mA/0sAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 3600x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(50,4))\n",
    "plt.plot(train.index,train['counts'],label='Train');\n",
    "plt.plot(test.index,test['counts'],label='test')\n",
    "plt.legend();"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "coastal-robertson",
   "metadata": {},
   "outputs": [],
   "source": [
    "y_col = 'counts'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "senior-northeast",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157571, 2) (157571, 1)\n"
     ]
    }
   ],
   "source": [
    "#separate X and y only for the train data (for now)\n",
    "X_train = train.drop(y_col,axis=1).copy()\n",
    "y_train = train[[y_col]].copy() # the double brakets here are to keep the y in a dataframe format, otherwise it will be pandas Series\n",
    "print(X_train.shape, y_train.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "central-handling",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(157571, 2)\n",
      "(157571, 1)\n",
      "(157571,)\n"
     ]
    }
   ],
   "source": [
    "Xscaler = MinMaxScaler(feature_range=(0, 1)) # scale so that all the X data will range from 0 to 1\n",
    "Xscaler.fit(X_train)\n",
    "scaled_X_train = Xscaler.transform(X_train)\n",
    "print(X_train.shape)\n",
    "Yscaler = MinMaxScaler(feature_range=(0, 1))\n",
    "Yscaler.fit(y_train)\n",
    "scaled_y_train = Yscaler.transform(y_train)\n",
    "print(scaled_y_train.shape)\n",
    "scaled_y_train = scaled_y_train.reshape(-1) # remove the second dimension from y so the shape changes from (n,1) to (n,)\n",
    "print(scaled_y_train.shape)\n",
    "\n",
    "scaled_y_train = np.insert(scaled_y_train, 0, 0)\n",
    "scaled_y_train = np.delete(scaled_y_train, -1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "organized-window",
   "metadata": {},
   "outputs": [],
   "source": [
    "tmpXtrainDf = pd.DataFrame(scaled_X_train, columns=['datetime', 'zone'])\n",
    "tmpXtrainDf['datetime'] = X_train['datetime']\n",
    "scaled_X_train2 = tmpXtrainDf.to_numpy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "swiss-zambia",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(32, 25, 2)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "n_input = 25 #how many samples/rows/timesteps to look in the past in order to forecast the next sample\n",
    "n_features= X_train.shape[1] # how many predictors/Xs/features we have to predict y\n",
    "b_size = 32 # Number of timeseries samples in each batch\n",
    "generator = TimeseriesGenerator(scaled_X_train, scaled_y_train, length=n_input, batch_size=b_size)\n",
    "\n",
    "print(generator[0][0].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "aggressive-culture",
   "metadata": {},
   "outputs": [],
   "source": [
    "x_train = X_train.to_numpy().reshape(1, X_train.shape[0], X_train.shape[1]) #ok shape (1,steps,5) - 1 sequence, many steps, 5 features\n",
    "y_train = y_train.to_numpy().reshape(1, y_train.shape[0], y_train.shape[1])\n",
    "# x_test = X_test.to_numpy().reshape(1, X_test.shape[0], X_test.shape[1])\n",
    "# y_test = Y_test.to_numpy().reshape(1, Y_test.shape[0], Y_test.shape[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "every-paradise",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-f8fbd4c101e4>\", line 4, in <module>\n",
      "    model.add(Dense(1), activation='relu')\n",
      "TypeError: add() got an unexpected keyword argument 'activation'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-f8fbd4c101e4>\", line 4, in <module>\n",
      "    model.add(Dense(1), activation='relu')\n",
      "TypeError: add() got an unexpected keyword argument 'activation'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-19-f8fbd4c101e4>\", line 4, in <module>\n",
      "    model.add(Dense(1), activation='relu')\n",
      "TypeError: add() got an unexpected keyword argument 'activation'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "model = Sequential()\n",
    "model.add(LSTM((20), input_shape=(n_input, n_features), return_sequences=True))\n",
    "model.add(LSTM(20, return_sequences=False))\n",
    "model.add(Dense(1, activation='relu'))\n",
    "model.compile(loss='mse', optimizer='adam')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "nutritional-success",
   "metadata": {},
   "outputs": [],
   "source": [
    "# from keras.datasets import mnist"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "separate-execution",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 952\n"
     ]
    }
   ],
   "source": [
    "# (X_train, y_train),(X_test, y_test) = mnist.load_data() "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "timely-conjunction",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "worth-pilot",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-6fc6bf792009>\", line 3, in <module>\n",
      "    model.fit_generator(generator, epochs=50)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\legacy\\interfaces.py\", line 91, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 1732, in fit_generator\n",
      "    initial_epoch=initial_epoch)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training_generator.py\", line 42, in fit_generator\n",
      "    model._make_train_function()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 303, in _make_train_function\n",
      "    raise RuntimeError('You must compile your model before using it.')\n",
      "RuntimeError: You must compile your model before using it.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-6fc6bf792009>\", line 3, in <module>\n",
      "    model.fit_generator(generator, epochs=50)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\legacy\\interfaces.py\", line 91, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 1732, in fit_generator\n",
      "    initial_epoch=initial_epoch)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training_generator.py\", line 42, in fit_generator\n",
      "    model._make_train_function()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 303, in _make_train_function\n",
      "    raise RuntimeError('You must compile your model before using it.')\n",
      "RuntimeError: You must compile your model before using it.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-21-6fc6bf792009>\", line 3, in <module>\n",
      "    model.fit_generator(generator, epochs=50)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\legacy\\interfaces.py\", line 91, in wrapper\n",
      "    return func(*args, **kwargs)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 1732, in fit_generator\n",
      "    initial_epoch=initial_epoch)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training_generator.py\", line 42, in fit_generator\n",
      "    model._make_train_function()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\keras\\engine\\training.py\", line 303, in _make_train_function\n",
      "    raise RuntimeError('You must compile your model before using it.')\n",
      "RuntimeError: You must compile your model before using it.\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'RuntimeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "# with tf.device(\"cpu:0\"):\n",
    "with tf.device(\"cpu:0\"):\n",
    "    model.fit_generator(generator, epochs=50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "global-midnight",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 953\n"
     ]
    }
   ],
   "source": [
    "model.save('../data/picklerCheckpoints/LSTM/relu50')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "swedish-flower",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-22-979fc70dc38f>\", line 1, in <module>\n",
      "    loss_per_epoch = model.history.history['loss']\n",
      "AttributeError: 'Sequential' object has no attribute 'history'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-22-979fc70dc38f>\", line 1, in <module>\n",
      "    loss_per_epoch = model.history.history['loss']\n",
      "AttributeError: 'Sequential' object has no attribute 'history'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-22-979fc70dc38f>\", line 1, in <module>\n",
      "    loss_per_epoch = model.history.history['loss']\n",
      "AttributeError: 'Sequential' object has no attribute 'history'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'AttributeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "loss_per_epoch = model.history.history['loss']\n",
    "plt.plot(range(len(loss_per_epoch)),loss_per_epoch);"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "instructional-nevada",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(32, 25, 2)\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 954\n"
     ]
    }
   ],
   "source": [
    "X_test = test.drop(y_col,axis=1).copy()\n",
    "scaled_X_test = Xscaler.transform(X_test)\n",
    "test_generator = TimeseriesGenerator(scaled_X_test, np.zeros(len(X_test)), length=n_input, batch_size=b_size)\n",
    "print(test_generator[0][0].shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "coupled-exposure",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-73c47ab90b6c>\", line 4, in <module>\n",
      "    results = pd.DataFrame({'y_true':test[y_col].values[n_input:],'y_pred':y_pred.ravel()})\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\frame.py\", line 529, in __init__\n",
      "    mgr = init_dict(data, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 287, in init_dict\n",
      "    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 80, in arrays_to_mgr\n",
      "    index = extract_index(arrays)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 401, in extract_index\n",
      "    raise ValueError(\"arrays must all be same length\")\n",
      "ValueError: arrays must all be same length\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ValueError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-73c47ab90b6c>\", line 4, in <module>\n",
      "    results = pd.DataFrame({'y_true':test[y_col].values[n_input:],'y_pred':y_pred.ravel()})\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\frame.py\", line 529, in __init__\n",
      "    mgr = init_dict(data, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 287, in init_dict\n",
      "    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 80, in arrays_to_mgr\n",
      "    index = extract_index(arrays)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 401, in extract_index\n",
      "    raise ValueError(\"arrays must all be same length\")\n",
      "ValueError: arrays must all be same length\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ValueError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-73c47ab90b6c>\", line 4, in <module>\n",
      "    results = pd.DataFrame({'y_true':test[y_col].values[n_input:],'y_pred':y_pred.ravel()})\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\frame.py\", line 529, in __init__\n",
      "    mgr = init_dict(data, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 287, in init_dict\n",
      "    return arrays_to_mgr(arrays, data_names, index, columns, dtype=dtype)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 80, in arrays_to_mgr\n",
      "    index = extract_index(arrays)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\pandas\\core\\internals\\construction.py\", line 401, in extract_index\n",
      "    raise ValueError(\"arrays must all be same length\")\n",
      "ValueError: arrays must all be same length\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'ValueError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "with tf.device(\"cpu:0\"):\n",
    "    y_pred_scaled = model.predict(test_generator)\n",
    "y_pred = Yscaler.inverse_transform(y_pred_scaled)\n",
    "results = pd.DataFrame({'y_true':test[y_col].values[n_input:],'y_pred':y_pred.ravel()})\n",
    "print(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "comprehensive-entry",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n",
      "ERROR:root:Internal Python error in the inspect module.\n",
      "Below is the traceback from this internal error.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-fae63fe7f9c9>\", line 1, in <module>\n",
      "    results.plot(alpha=0.7)\n",
      "NameError: name 'results' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "ERROR! Session/line number was not unique in database. History logging moved to new session 955\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-fae63fe7f9c9>\", line 1, in <module>\n",
      "    results.plot(alpha=0.7)\n",
      "NameError: name 'results' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3427, in run_code\n",
      "    exec(code_obj, self.user_global_ns, self.user_ns)\n",
      "  File \"<ipython-input-23-fae63fe7f9c9>\", line 1, in <module>\n",
      "    results.plot(alpha=0.7)\n",
      "NameError: name 'results' is not defined\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'NameError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3347, in run_ast_nodes\n",
      "    if (await self.run_code(code, result,  async_=asy)):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3444, in run_code\n",
      "    self.showtraceback(running_compiled_code=True)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1125, in structured_traceback\n",
      "    tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2932, in _run_cell\n",
      "    return runner(coro)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\async_helpers.py\", line 68, in _pseudo_sync_runner\n",
      "    coro.send(None)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3156, in run_cell_async\n",
      "    interactivity=interactivity, compiler=compiler, result=result)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 3366, in run_ast_nodes\n",
      "    self.showtraceback()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2057, in showtraceback\n",
      "    value, tb, tb_offset=tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1368, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1268, in structured_traceback\n",
      "    self, etype, value, tb, tb_offset, number_of_lines_of_context\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1143, in structured_traceback\n",
      "    chained_exceptions_tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1082, in format_exception_as_a_whole\n",
      "    last_unique, recursion_repeat = find_recursion(orig_etype, evalue, records)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 382, in find_recursion\n",
      "    return len(records), 0\n",
      "TypeError: object of type 'NoneType' has no len()\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\interactiveshell.py\", line 2054, in showtraceback\n",
      "    stb = value._render_traceback_()\n",
      "AttributeError: 'TypeError' object has no attribute '_render_traceback_'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 1101, in get_records\n",
      "    return _fixed_getinnerframes(etb, number_of_lines_of_context, tb_offset)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 248, in wrapped\n",
      "    return f(*args, **kwargs)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\IPython\\core\\ultratb.py\", line 281, in _fixed_getinnerframes\n",
      "    records = fix_frame_records_filenames(inspect.getinnerframes(etb, context))\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1502, in getinnerframes\n",
      "    frameinfo = (tb.tb_frame,) + getframeinfo(tb, context)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 1460, in getframeinfo\n",
      "    filename = getsourcefile(frame) or getfile(frame)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 696, in getsourcefile\n",
      "    if getattr(getmodule(object, filename), '__loader__', None) is not None:\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\inspect.py\", line 733, in getmodule\n",
      "    if ismodule(module) and hasattr(module, '__file__'):\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 50, in __getattr__\n",
      "    module = self._load()\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\site-packages\\tensorflow\\__init__.py\", line 44, in _load\n",
      "    module = _importlib.import_module(self.__name__)\n",
      "  File \"D:\\Anaconda\\envs\\tensorflow\\lib\\importlib\\__init__.py\", line 127, in import_module\n",
      "    return _bootstrap._gcd_import(name[level:], package, level)\n",
      "  File \"<frozen importlib._bootstrap>\", line 1006, in _gcd_import\n",
      "  File \"<frozen importlib._bootstrap>\", line 983, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 965, in _find_and_load_unlocked\n",
      "ModuleNotFoundError: No module named 'tensorflow_core.estimator'\n"
     ]
    }
   ],
   "source": [
    "results.plot(alpha=0.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "scientific-polish",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 16.82662  , -38.405003 ,  51.34356  , ..., -36.995224 ,\n",
       "        -10.704786 ,   7.343804 ],\n",
       "       [ 16.85086  , -38.60882  ,  51.59898  , ..., -36.69274  ,\n",
       "        -11.279827 ,   6.7438774],\n",
       "       [ 16.876612 , -38.80567  ,  51.854057 , ..., -36.383232 ,\n",
       "        -11.838027 ,   6.151756 ],\n",
       "       ...,\n",
       "       [ 18.524124 , -44.89713  ,  57.92525  , ..., -31.942688 ,\n",
       "        -25.458822 ,  -2.745086 ],\n",
       "       [ 18.514713 , -45.011566 ,  58.063473 , ..., -31.719799 ,\n",
       "        -25.862774 ,  -3.0857232],\n",
       "       [ 18.510727 , -45.12314  ,  58.201107 , ..., -31.492552 ,\n",
       "        -26.257555 ,  -3.4243274]], dtype=float32)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR! Session/line number was not unique in database. History logging moved to new session 956\n"
     ]
    }
   ],
   "source": [
    "y_pred"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.7 (tensorflow)",
   "language": "python",
   "name": "tensorflow"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
